{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "43a93c76-e5f0-4b20-8e80-aabddb4f6034",
   "metadata": {},
   "source": [
    "<p>We will install de software and dependencies using the Conda Environment.</p>\n",
    "<p>First, we will update our conda environment:</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b8cea73d-fbed-4888-849f-ad3f901e557f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - conda\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    brotlipy-0.7.0             |py37h5e8e339_1003         342 KB  conda-forge\n",
      "    chardet-4.0.0              |   py37h89c1867_2         199 KB  conda-forge\n",
      "    charset-normalizer-2.0.9   |     pyhd8ed1ab_0          34 KB  conda-forge\n",
      "    conda-package-handling-1.7.3|   py37h5e8e339_1        1008 KB  conda-forge\n",
      "    cookiecutter-1.7.3         |     pyh6c4a22f_0          82 KB  conda-forge\n",
      "    cryptography-36.0.0        |   py37hf1a17b8_0         1.7 MB  conda-forge\n",
      "    docker-py-5.0.3            |   py37h89c1867_2         189 KB  conda-forge\n",
      "    fonttools-4.28.3           |   py37h5e8e339_0         1.6 MB  conda-forge\n",
      "    glib-2.70.2                |       h780b84a_0         438 KB  conda-forge\n",
      "    glib-tools-2.70.2          |       h780b84a_0         106 KB  conda-forge\n",
      "    google-api-python-client-2.33.0|     pyhd8ed1ab_0         4.3 MB  conda-forge\n",
      "    google-cloud-storage-1.43.0|     pyh6c4a22f_0          73 KB  conda-forge\n",
      "    hdf5-1.12.1                |nompi_h2750804_103         3.6 MB  conda-forge\n",
      "    ipykernel-6.6.0            |   py37h6531663_0         177 KB  conda-forge\n",
      "    ipython-7.30.1             |   py37h89c1867_0         1.1 MB  conda-forge\n",
      "    jinja2-2.11.3              |     pyh44b312d_0          93 KB  conda-forge\n",
      "    jsonschema-4.2.1           |     pyhd8ed1ab_1          55 KB  conda-forge\n",
      "    jupyter_server-1.13.0      |     pyhd8ed1ab_0         264 KB  conda-forge\n",
      "    jupytext-1.13.3            |     pyheef035f_0         196 KB  conda-forge\n",
      "    libglib-2.70.2             |       h174f98d_0         3.1 MB  conda-forge\n",
      "    libopencv-4.5.3            |   py37hbfc4018_6        34.5 MB  conda-forge\n",
      "    libpq-13.5                 |       hd57d9b9_1         2.8 MB  conda-forge\n",
      "    markupsafe-1.1.1           |   py37h5e8e339_3          27 KB  conda-forge\n",
      "    matplotlib-3.5.0           |   py37h89c1867_0           6 KB  conda-forge\n",
      "    ncurses-6.2                |       h58526e2_4         985 KB  conda-forge\n",
      "    nss-3.73                   |       hb5efdd6_0         2.1 MB  conda-forge\n",
      "    opencv-4.5.3               |   py37h89c1867_6          22 KB  conda-forge\n",
      "    pandas-profiling-1.4.1     |                0          39 KB  conda-forge\n",
      "    prompt-toolkit-3.0.24      |     pyha770c72_0         249 KB  conda-forge\n",
      "    py-opencv-4.5.3            |   py37h6531663_6         1.1 MB  conda-forge\n",
      "    pycosat-0.6.3              |py37h5e8e339_1009         107 KB  conda-forge\n",
      "    pyjwt-2.3.0                |     pyhd8ed1ab_1          18 KB  conda-forge\n",
      "    pyqt-5.12.3                |   py37h89c1867_8          22 KB  conda-forge\n",
      "    pyqt-impl-5.12.3           |   py37hac37412_8         5.9 MB  conda-forge\n",
      "    pyqt5-sip-4.19.18          |   py37hcd2ae1e_8         310 KB  conda-forge\n",
      "    pyqtchart-5.12             |   py37he336c9b_8         257 KB  conda-forge\n",
      "    pyqtwebengine-5.12.1       |   py37he336c9b_8         175 KB  conda-forge\n",
      "    pysocks-1.7.1              |   py37h89c1867_4          28 KB  conda-forge\n",
      "    readline-8.1               |       h46c0cb4_0         295 KB  conda-forge\n",
      "    ruamel_yaml-0.15.80        |py37h5e8e339_1006         270 KB  conda-forge\n",
      "    six-1.16.0                 |     pyh6c4a22f_0          14 KB  conda-forge\n",
      "    typed-ast-1.5.1            |   py37h5e8e339_0         208 KB  conda-forge\n",
      "    uritemplate-4.1.1          |     pyhd8ed1ab_0          12 KB  conda-forge\n",
      "    visions-0.7.5              |     pyhd8ed1ab_0          59 KB  conda-forge\n",
      "    websocket-client-1.2.3     |     pyhd8ed1ab_0          41 KB  conda-forge\n",
      "    xz-5.2.5                   |       h516909a_1         343 KB  conda-forge\n",
      "    yaml-0.2.5                 |       h516909a_0          82 KB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        68.5 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  matplotlib         conda-forge/linux-64::matplotlib-3.5.0-py37h89c1867_0\n",
      "  pyqt               conda-forge/linux-64::pyqt-5.12.3-py37h89c1867_8\n",
      "  pyqt-impl          conda-forge/linux-64::pyqt-impl-5.12.3-py37hac37412_8\n",
      "  pyqt5-sip          conda-forge/linux-64::pyqt5-sip-4.19.18-py37hcd2ae1e_8\n",
      "  pyqtchart          conda-forge/linux-64::pyqtchart-5.12-py37he336c9b_8\n",
      "  pyqtwebengine      conda-forge/linux-64::pyqtwebengine-5.12.1-py37he336c9b_8\n",
      "\n",
      "The following packages will be REMOVED:\n",
      "\n",
      "  future-0.18.2-py37h89c1867_4\n",
      "  whichcraft-0.6.1-py_0\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  charset-normalizer                     2.0.8-pyhd8ed1ab_0 --> 2.0.9-pyhd8ed1ab_0\n",
      "  cookiecutter                                   1.7.0-py_0 --> 1.7.3-pyh6c4a22f_0\n",
      "  cryptography                        35.0.0-py37hf1a17b8_2 --> 36.0.0-py37hf1a17b8_0\n",
      "  docker-py                            5.0.3-py37h89c1867_0 --> 5.0.3-py37h89c1867_2\n",
      "  fonttools                           4.28.2-py37h5e8e339_0 --> 4.28.3-py37h5e8e339_0\n",
      "  glib                                    2.70.1-h780b84a_0 --> 2.70.2-h780b84a_0\n",
      "  glib-tools                              2.70.1-h780b84a_0 --> 2.70.2-h780b84a_0\n",
      "  google-api-python~                    2.31.0-pyhd8ed1ab_0 --> 2.33.0-pyhd8ed1ab_0\n",
      "  google-cloud-stor~                    1.42.3-pyh6c4a22f_1 --> 1.43.0-pyh6c4a22f_0\n",
      "  hdf5                            1.12.1-nompi_h2750804_102 --> 1.12.1-nompi_h2750804_103\n",
      "  ipykernel                            6.5.1-py37h6531663_0 --> 6.6.0-py37h6531663_0\n",
      "  ipython                             7.30.0-py37h89c1867_0 --> 7.30.1-py37h89c1867_0\n",
      "  jsonschema                             4.2.1-pyhd8ed1ab_0 --> 4.2.1-pyhd8ed1ab_1\n",
      "  jupyter_server                        1.12.1-pyhd8ed1ab_0 --> 1.13.0-pyhd8ed1ab_0\n",
      "  jupytext                              1.13.2-pyh6002c4b_0 --> 1.13.3-pyheef035f_0\n",
      "  libglib                                 2.70.1-h174f98d_0 --> 2.70.2-h174f98d_0\n",
      "  libopencv                            4.5.3-py37hbfc4018_5 --> 4.5.3-py37hbfc4018_6\n",
      "  libpq                                     13.5-hd57d9b9_0 --> 13.5-hd57d9b9_1\n",
      "  ncurses                 pkgs/main::ncurses-6.2-he6710b0_1 --> conda-forge::ncurses-6.2-h58526e2_4\n",
      "  nss                                       3.72-hb5efdd6_0 --> 3.73-hb5efdd6_0\n",
      "  opencv                               4.5.3-py37h89c1867_5 --> 4.5.3-py37h89c1867_6\n",
      "  prompt-toolkit                        3.0.22-pyha770c72_0 --> 3.0.24-pyha770c72_0\n",
      "  py-opencv                            4.5.3-py37h6531663_5 --> 4.5.3-py37h6531663_6\n",
      "  pycosat            pkgs/main::pycosat-0.6.3-py37h27cfd23~ --> conda-forge::pycosat-0.6.3-py37h5e8e339_1009\n",
      "  pyjwt                                  2.3.0-pyhd8ed1ab_0 --> 2.3.0-pyhd8ed1ab_1\n",
      "  pysocks                   pkgs/main::pysocks-1.7.1-py37_1 --> conda-forge::pysocks-1.7.1-py37h89c1867_4\n",
      "  typed-ast                            1.5.0-py37h5e8e339_0 --> 1.5.1-py37h5e8e339_0\n",
      "  uritemplate                                    3.0.1-py_0 --> 4.1.1-pyhd8ed1ab_0\n",
      "  visions                                0.7.1-pyhd8ed1ab_0 --> 0.7.5-pyhd8ed1ab_0\n",
      "  websocket-client   conda-forge/linux-64::websocket-clien~ --> conda-forge/noarch::websocket-client-1.2.3-pyhd8ed1ab_0\n",
      "  xz                         pkgs/main::xz-5.2.5-h7b6447c_0 --> conda-forge::xz-5.2.5-h516909a_1\n",
      "\n",
      "The following packages will be SUPERSEDED by a higher-priority channel:\n",
      "\n",
      "  brotlipy           pkgs/main::brotlipy-0.7.0-py37h27cfd2~ --> conda-forge::brotlipy-0.7.0-py37h5e8e339_1003\n",
      "  chardet            pkgs/main::chardet-4.0.0-py37h06a4308~ --> conda-forge::chardet-4.0.0-py37h89c1867_2\n",
      "  conda-package-han~ pkgs/main::conda-package-handling-1.7~ --> conda-forge::conda-package-handling-1.7.3-py37h5e8e339_1\n",
      "  pandas-profiling   conda-forge/noarch::pandas-profiling-~ --> conda-forge/linux-64::pandas-profiling-1.4.1-0\n",
      "  readline               pkgs/main::readline-8.1-h27cfd23_0 --> conda-forge::readline-8.1-h46c0cb4_0\n",
      "  ruamel_yaml        pkgs/main::ruamel_yaml-0.15.100-py37h~ --> conda-forge::ruamel_yaml-0.15.80-py37h5e8e339_1006\n",
      "  six                    pkgs/main::six-1.16.0-pyhd3eb1b0_0 --> conda-forge::six-1.16.0-pyh6c4a22f_0\n",
      "  yaml                     pkgs/main::yaml-0.2.5-h7b6447c_0 --> conda-forge::yaml-0.2.5-h516909a_0\n",
      "\n",
      "The following packages will be DOWNGRADED:\n",
      "\n",
      "  jinja2                                 3.0.3-pyhd8ed1ab_0 --> 2.11.3-pyh44b312d_0\n",
      "  markupsafe                           2.0.1-py37h5e8e339_1 --> 1.1.1-py37h5e8e339_3\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "jinja2-2.11.3        | 93 KB     | ##################################### | 100% \n",
      "readline-8.1         | 295 KB    | ##################################### | 100% \n",
      "pyqtwebengine-5.12.1 | 175 KB    | ##################################### | 100% \n",
      "six-1.16.0           | 14 KB     | ##################################### | 100% \n",
      "charset-normalizer-2 | 34 KB     | ##################################### | 100% \n",
      "google-api-python-cl | 4.3 MB    | ##################################### | 100% \n",
      "xz-5.2.5             | 343 KB    | ##################################### | 100% \n",
      "pyjwt-2.3.0          | 18 KB     | ##################################### | 100% \n",
      "prompt-toolkit-3.0.2 | 249 KB    | ##################################### | 100% \n",
      "jupytext-1.13.3      | 196 KB    | ##################################### | 100% \n",
      "ipykernel-6.6.0      | 177 KB    | ##################################### | 100% \n",
      "nss-3.73             | 2.1 MB    | ##################################### | 100% \n",
      "pysocks-1.7.1        | 28 KB     | ##################################### | 100% \n",
      "pandas-profiling-1.4 | 39 KB     | ##################################### | 100% \n",
      "brotlipy-0.7.0       | 342 KB    | ##################################### | 100% \n",
      "libpq-13.5           | 2.8 MB    | ##################################### | 100% \n",
      "chardet-4.0.0        | 199 KB    | ##################################### | 100% \n",
      "ipython-7.30.1       | 1.1 MB    | ##################################### | 100% \n",
      "yaml-0.2.5           | 82 KB     | ##################################### | 100% \n",
      "fonttools-4.28.3     | 1.6 MB    | ##################################### | 100% \n",
      "pyqt-impl-5.12.3     | 5.9 MB    | ##################################### | 100% \n",
      "visions-0.7.5        | 59 KB     | ##################################### | 100% \n",
      "uritemplate-4.1.1    | 12 KB     | ##################################### | 100% \n",
      "pyqtchart-5.12       | 257 KB    | ##################################### | 100% \n",
      "libglib-2.70.2       | 3.1 MB    | ##################################### | 100% \n",
      "glib-tools-2.70.2    | 106 KB    | ##################################### | 100% \n",
      "typed-ast-1.5.1      | 208 KB    | ##################################### | 100% \n",
      "google-cloud-storage | 73 KB     | ##################################### | 100% \n",
      "conda-package-handli | 1008 KB   | ##################################### | 100% \n",
      "ncurses-6.2          | 985 KB    | ##################################### | 100% \n",
      "py-opencv-4.5.3      | 1.1 MB    | ##################################### | 100% \n",
      "glib-2.70.2          | 438 KB    | ##################################### | 100% \n",
      "docker-py-5.0.3      | 189 KB    | ##################################### | 100% \n",
      "pyqt5-sip-4.19.18    | 310 KB    | ##################################### | 100% \n",
      "cryptography-36.0.0  | 1.7 MB    | ##################################### | 100% \n",
      "matplotlib-3.5.0     | 6 KB      | ##################################### | 100% \n",
      "opencv-4.5.3         | 22 KB     | ##################################### | 100% \n",
      "websocket-client-1.2 | 41 KB     | ##################################### | 100% \n",
      "pyqt-5.12.3          | 22 KB     | ##################################### | 100% \n",
      "jsonschema-4.2.1     | 55 KB     | ##################################### | 100% \n",
      "ruamel_yaml-0.15.80  | 270 KB    | ##################################### | 100% \n",
      "libopencv-4.5.3      | 34.5 MB   | ##################################### | 100% \n",
      "jupyter_server-1.13. | 264 KB    | ##################################### | 100% \n",
      "hdf5-1.12.1          | 3.6 MB    | ##################################### | 100% \n",
      "cookiecutter-1.7.3   | 82 KB     | ##################################### | 100% \n",
      "pycosat-0.6.3        | 107 KB    | ##################################### | 100% \n",
      "markupsafe-1.1.1     | 27 KB     | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda update -n base conda"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "903a8231-2e6f-4686-be2e-742a8a81b70c",
   "metadata": {},
   "source": [
    "<p>Install de <strong>Bioconda&nbsp;</strong>package, which harbors most of our pipeline&apos;s dependencies:</p>\n",
    "<p><a href=\"https://bioconda.github.io/user/install.html\">https://bioconda.github.io/user/install.html</a></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "385754e7-91d1-4f56-b587-a9acf9c7ef7c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda config --add channels bioconda"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6d528ec-4525-4fd1-acad-2f34507042be",
   "metadata": {},
   "source": [
    "<p>Install <strong>Cutadapt</strong>, which is the tool adopted to run the demultiplexing step and adapter, tags and primers removal.</p>\n",
    "<p><a href=\"https://cutadapt.readthedocs.io/en/stable/index.html\">https://cutadapt.readthedocs.io/en/stable/index.html</a></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8657fc19-6aef-44ff-9a70-94edf8e1f0c8",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - cutadapt\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    cutadapt-3.5               |   py37h73a75cf_0         208 KB  bioconda\n",
      "    dnaio-0.6.0                |   py37h73a75cf_0         140 KB  bioconda\n",
      "    isa-l-2.30.0               |       ha770c72_4         192 KB  conda-forge\n",
      "    pbzip2-1.1.13              |                0         114 KB  conda-forge\n",
      "    pigz-2.6                   |       h27826a3_0          87 KB  conda-forge\n",
      "    python-isal-0.11.1         |   py37h5e8e339_1         136 KB  conda-forge\n",
      "    xopen-1.2.1                |   py37h89c1867_1          23 KB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         899 KB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  cutadapt           bioconda/linux-64::cutadapt-3.5-py37h73a75cf_0\n",
      "  dnaio              bioconda/linux-64::dnaio-0.6.0-py37h73a75cf_0\n",
      "  isa-l              conda-forge/linux-64::isa-l-2.30.0-ha770c72_4\n",
      "  pbzip2             conda-forge/linux-64::pbzip2-1.1.13-0\n",
      "  pigz               conda-forge/linux-64::pigz-2.6-h27826a3_0\n",
      "  python-isal        conda-forge/linux-64::python-isal-0.11.1-py37h5e8e339_1\n",
      "  xopen              conda-forge/linux-64::xopen-1.2.1-py37h89c1867_1\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "pbzip2-1.1.13        | 114 KB    | ##################################### | 100% \n",
      "dnaio-0.6.0          | 140 KB    | ##################################### | 100% \n",
      "cutadapt-3.5         | 208 KB    | ##################################### | 100% \n",
      "isa-l-2.30.0         | 192 KB    | ##################################### | 100% \n",
      "python-isal-0.11.1   | 136 KB    | ##################################### | 100% \n",
      "pigz-2.6             | 87 KB     | ##################################### | 100% \n",
      "xopen-1.2.1          | 23 KB     | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c bioconda cutadapt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82a0ede4-6eef-4ec3-8171-8a4ff3cd21de",
   "metadata": {},
   "source": [
    "Check the instalation by asking for Cutadapt help:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a1dfc055-b27a-4861-9d8e-fb8652245833",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cutadapt version 3.5\n",
      "\n",
      "Copyright (C) 2010-2021 Marcel Martin <marcel.martin@scilifelab.se>\n",
      "\n",
      "cutadapt removes adapter sequences from high-throughput sequencing reads.\n",
      "\n",
      "Usage:\n",
      "    cutadapt -a ADAPTER [options] [-o output.fastq] input.fastq\n",
      "\n",
      "For paired-end reads:\n"
     ]
    }
   ],
   "source": [
    "!cutadapt -h | head"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32eb417e-89f7-41dd-a336-b5e7fdc14416",
   "metadata": {},
   "source": [
    "<p>Install <strong>VSEARCH</strong>, which is the tool adopted for read cleaning and sequences headers formatting</p>\n",
    "<p><a href=\"https://github.com/torognes/vsearch\">https://github.com/torognes/vsearch</a></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7fda2a13-aa08-4dd8-b761-1b98e4790008",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - vsearch\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    vsearch-2.18.0             |       h95f258a_0         1.4 MB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         1.4 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  vsearch            bioconda/linux-64::vsearch-2.18.0-h95f258a_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "vsearch-2.18.0       | 1.4 MB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c bioconda vsearch"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86b29988-f86c-4a24-8e12-b7971de5607b",
   "metadata": {},
   "source": [
    "Check the instalation by asking for VSEARCH help:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "60b09162-ce63-4900-a2b1-7141d5224dcc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "vsearch v2.18.0_linux_x86_64, 14.7GB RAM, 4 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Rognes T, Flouri T, Nichols B, Quince C, Mahe F (2016)\n",
      "VSEARCH: a versatile open source tool for metagenomics\n",
      "PeerJ 4:e2584 doi: 10.7717/peerj.2584 https://doi.org/10.7717/peerj.2584\n",
      "\n",
      "Usage: vsearch [OPTIONS]\n",
      "\n",
      "General options\n",
      "  --bzip2_decompress          decompress input with bzip2 (required if pipe)\n",
      "  --fasta_width INT           width of FASTA seq lines, 0 for no wrap (80)\n",
      "  --gzip_decompress           decompress input with gzip (required if pipe)\n"
     ]
    }
   ],
   "source": [
    "!vsearch -h | head"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1885b67b-a0fd-4bd5-8354-b3baf4a31789",
   "metadata": {},
   "source": [
    "<p>Install <strong>SeqKit</strong>, which is the tool adopted for references databases formatting</p>\n",
    "<p><a href=\"https://bioinf.shenwei.me/seqkit/\">https://bioinf.shenwei.me/seqkit/</a></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "98fd56ab-af96-4b44-98a4-e06971d885b9",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - seqkit\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    seqkit-2.1.0               |       h9ee0642_0         6.1 MB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         6.1 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  seqkit             bioconda/linux-64::seqkit-2.1.0-h9ee0642_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "seqkit-2.1.0         | 6.1 MB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c bioconda seqkit"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5550d77e-8894-4483-84de-972fc8a5610e",
   "metadata": {},
   "source": [
    "Check the instalation by asking for seqkit help:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b9d30a74-aa1e-46ca-8be1-9ffda8d24d53",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SeqKit -- a cross-platform and ultrafast toolkit for FASTA/Q file manipulation\n",
      "\n",
      "Version: 2.1.0\n",
      "\n",
      "Author: Wei Shen <shenwei356@gmail.com>\n",
      "\n",
      "Documents  : http://bioinf.shenwei.me/seqkit\n",
      "Source code: https://github.com/shenwei356/seqkit\n",
      "Please cite: https://doi.org/10.1371/journal.pone.0163962\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!seqkit -h | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "34113365-1259-4625-b651-021215eeab36",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - seqtk\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    seqtk-1.3                  |       h5bf99c6_3          38 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:          38 KB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  seqtk              bioconda/linux-64::seqtk-1.3-h5bf99c6_3\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "seqtk-1.3            | 38 KB     | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c bioconda seqtk"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c10a33c-1076-4fa8-a58a-22d5b2dc1b78",
   "metadata": {},
   "source": [
    "<p>Install <strong>BioPerl&nbsp;</strong>package, which is used for removing sequences duplicates from the references databases</p>\n",
    "<p><a href=\"https://bioperl.org/\">https://bioperl.org/</a></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "69c3d6c4-bd71-443e-be38-3f5d0dd6f1c5",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - perl-bioperl\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    libgcc-7.2.0               |       h69d50b8_2         304 KB  conda-forge\n",
      "    perl-5.26.2                |    h36c2ea0_1008        15.4 MB  conda-forge\n",
      "    perl-bioperl-1.6.924       |                4         2.2 MB  bioconda\n",
      "    perl-threaded-5.26.0       |                0           4 KB  bioconda\n",
      "    perl-yaml-1.29             |          pl526_0          41 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        17.9 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  libgcc             conda-forge/linux-64::libgcc-7.2.0-h69d50b8_2\n",
      "  perl               conda-forge/linux-64::perl-5.26.2-h36c2ea0_1008\n",
      "  perl-bioperl       bioconda/linux-64::perl-bioperl-1.6.924-4\n",
      "  perl-threaded      bioconda/linux-64::perl-threaded-5.26.0-0\n",
      "  perl-yaml          bioconda/noarch::perl-yaml-1.29-pl526_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "perl-threaded-5.26.0 | 4 KB      | ##################################### | 100% \n",
      "perl-bioperl-1.6.924 | 2.2 MB    | ##################################### | 100% \n",
      "perl-5.26.2          | 15.4 MB   | ##################################### | 100% \n",
      "perl-yaml-1.29       | 41 KB     | ##################################### | 100% \n",
      "libgcc-7.2.0         | 304 KB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c bioconda perl-bioperl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e54f0d8d-6160-46ab-8445-c6937014ef60",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - perl-bioperl\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    munkres-1.0.7              |             py_1          10 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:          10 KB\n",
      "\n",
      "The following packages will be SUPERSEDED by a higher-priority channel:\n",
      "\n",
      "  munkres            conda-forge::munkres-1.1.4-pyh9f0ad1d~ --> bioconda::munkres-1.0.7-py_1\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "munkres-1.0.7        | 10 KB     | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda update perl-bioperl"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8321c049-7820-420a-a7a2-a2dfa3f8b268",
   "metadata": {},
   "source": [
    "<p>Install <strong>SWARM&nbsp;</strong>which is the tool adopted for clustering the amplicons into OTUs</p>\n",
    "<p><a href=\"https://github.com/frederic-mahe/swarm\">https://github.com/frederic-mahe/swarm</a></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ca6a4fc6-8396-4342-8525-bd24645868e6",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - biopython\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    biopython-1.78             |   py37h7b6447c_0         2.6 MB  anaconda\n",
      "    ca-certificates-2020.10.14 |                0         128 KB  anaconda\n",
      "    certifi-2020.6.20          |           py37_0         159 KB  anaconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         2.9 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  biopython          anaconda/linux-64::biopython-1.78-py37h7b6447c_0\n",
      "\n",
      "The following packages will be SUPERSEDED by a higher-priority channel:\n",
      "\n",
      "  ca-certificates    conda-forge::ca-certificates-2021.10.~ --> anaconda::ca-certificates-2020.10.14-0\n",
      "  certifi            conda-forge::certifi-2021.10.8-py37h8~ --> anaconda::certifi-2020.6.20-py37_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "biopython-1.78       | 2.6 MB    | ##################################### | 100% \n",
      "ca-certificates-2020 | 128 KB    | ##################################### | 100% \n",
      "certifi-2020.6.20    | 159 KB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c anaconda biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2a91f25e-e290-4c23-8935-d40895cdc8b8",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - snippy==4.05\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    any2fasta-0.4.2            |       hdfd78af_3           8 KB  bioconda\n",
      "    atk-1.0-2.36.0             |       h3371d22_4         560 KB  conda-forge\n",
      "    bcftools-1.14              |       h88f3f91_0         911 KB  bioconda\n",
      "    bedtools-2.30.0            |       h7d7f7ad_2        17.9 MB  bioconda\n",
      "    bwa-0.7.17                 |       h5bf99c6_8         608 KB  bioconda\n",
      "    clustalw-2.1               |       h7d875b9_6         318 KB  bioconda\n",
      "    curl-7.80.0                |       h2574ce0_0         153 KB  conda-forge\n",
      "    font-ttf-dejavu-sans-mono-2.37|       hab24e00_0         388 KB  conda-forge\n",
      "    font-ttf-inconsolata-3.000 |       h77eed37_0          94 KB  conda-forge\n",
      "    font-ttf-source-code-pro-2.038|       h77eed37_0         684 KB  conda-forge\n",
      "    font-ttf-ubuntu-0.83       |       hab24e00_0         1.9 MB  conda-forge\n",
      "    fonts-conda-ecosystem-1    |                0           4 KB  conda-forge\n",
      "    fonts-conda-forge-1        |                0           4 KB  conda-forge\n",
      "    freebayes-1.3.2            |   py37h26878c9_2        16.3 MB  bioconda\n",
      "    fribidi-1.0.10             |       h36c2ea0_0         112 KB  conda-forge\n",
      "    gdk-pixbuf-2.42.6          |       h04a7f16_0         609 KB  conda-forge\n",
      "    giflib-5.2.1               |       h36c2ea0_2          77 KB  conda-forge\n",
      "    graphviz-2.50.0            |       h85b4f2f_1         2.9 MB  conda-forge\n",
      "    gsl-2.6                    |       he838d99_2         3.2 MB  conda-forge\n",
      "    gtk2-2.24.33               |       h539f30e_1         7.3 MB  conda-forge\n",
      "    gts-0.7.6                  |       h64030ff_2         411 KB  conda-forge\n",
      "    htslib-1.14                |       h9093b5e_0         2.2 MB  bioconda\n",
      "    k8-0.2.5                   |       h9a82719_1         1.7 MB  bioconda\n",
      "    lerc-2.2.1                 |       h9c3ff4c_0         213 KB  conda-forge\n",
      "    libdb-6.1.26               |                0        21.5 MB  bioconda\n",
      "    libdeflate-1.7             |       h7f98852_5          67 KB  conda-forge\n",
      "    libgd-2.3.3                |       h6ad9fb6_0         279 KB  conda-forge\n",
      "    librsvg-2.52.4             |       hc3c00ef_0         5.3 MB  conda-forge\n",
      "    libtiff-4.3.0              |       hf544144_1         668 KB  conda-forge\n",
      "    libtool-2.4.6              |    h9c3ff4c_1008         511 KB  conda-forge\n",
      "    libwebp-1.2.1              |       h3452ae3_0          84 KB  conda-forge\n",
      "    libxslt-1.1.33             |       h15afd5d_2         522 KB  conda-forge\n",
      "    minimap2-2.23              |       h5bf99c6_0         489 KB  bioconda\n",
      "    openjdk-8.0.312            |       h7f98852_0        97.6 MB  conda-forge\n",
      "    paml-4.9                   |       h779adbc_6         1.2 MB  bioconda\n",
      "    pango-1.48.10              |       h54213e6_2         403 KB  conda-forge\n",
      "    parallel-20211022          |       ha770c72_0         1.8 MB  conda-forge\n",
      "    perl-aceperl-1.92          |          pl526_2         149 KB  bioconda\n",
      "    perl-algorithm-diff-1.1903 |          pl526_2          34 KB  bioconda\n",
      "    perl-algorithm-munkres-0.08|          pl526_1          11 KB  bioconda\n",
      "    perl-apache-test-1.40      |          pl526_1         115 KB  bioconda\n",
      "    perl-app-cpanminus-1.7044  |          pl526_1         234 KB  bioconda\n",
      "    perl-appconfig-1.71        |          pl526_1          42 KB  bioconda\n",
      "    perl-array-compare-3.0.1   |          pl526_1          12 KB  bioconda\n",
      "    perl-autoloader-5.74       |          pl526_2           5 KB  bioconda\n",
      "    perl-base-2.23             |          pl526_1          10 KB  bioconda\n",
      "    perl-bio-asn1-entrezgene-1.73| pl5262hdfd78af_2          23 KB  bioconda\n",
      "    perl-bio-coordinate-1.007001| pl5262hdfd78af_2          35 KB  bioconda\n",
      "    perl-bio-featureio-1.6.905 | pl5262hdfd78af_3          40 KB  bioconda\n",
      "    perl-bio-phylo-0.58        | pl5262hdfd78af_3         310 KB  bioconda\n",
      "    perl-bio-samtools-1.43     |  pl526h1341992_1         172 KB  bioconda\n",
      "    perl-bio-tools-phylo-paml-1.7.3| pl5262hdfd78af_2          57 KB  bioconda\n",
      "    perl-bio-tools-run-alignment-clustalw-1.7.4| pl5262hdfd78af_2          25 KB  bioconda\n",
      "    perl-bio-tools-run-alignment-tcoffee-1.7.4| pl5262hdfd78af_3          24 KB  bioconda\n",
      "    perl-bioperl-1.7.2         |         pl526_11          13 KB  bioconda\n",
      "    perl-bioperl-core-1.007002 | pl5262hdfd78af_3         3.2 MB  bioconda\n",
      "    perl-bioperl-run-1.007002  | pl5262hdfd78af_5         431 KB  bioconda\n",
      "    perl-business-isbn-3.004   |          pl526_0          15 KB  bioconda\n",
      "    perl-business-isbn-data-20140910.003|          pl526_0          15 KB  bioconda\n",
      "    perl-cache-cache-1.08      |          pl526_0          31 KB  bioconda\n",
      "    perl-capture-tiny-0.48     |          pl526_0          17 KB  bioconda\n",
      "    perl-carp-1.38             |          pl526_3          15 KB  bioconda\n",
      "    perl-cgi-4.44              |  pl526h14c3975_1         116 KB  bioconda\n",
      "    perl-class-data-inheritable-0.08|          pl526_1           7 KB  bioconda\n",
      "    perl-class-inspector-1.34  |          pl526_0          14 KB  bioconda\n",
      "    perl-class-load-0.25       |          pl526_0          12 KB  bioconda\n",
      "    perl-class-load-xs-0.10    |  pl526h6bb024c_2          13 KB  bioconda\n",
      "    perl-class-method-modifiers-2.12|          pl526_0          13 KB  bioconda\n",
      "    perl-clone-0.42            |  pl526h516909a_0          13 KB  bioconda\n",
      "    perl-common-sense-3.74     |          pl526_2          11 KB  bioconda\n",
      "    perl-compress-raw-zlib-2.087|  pl526hc9558a2_0          72 KB  bioconda\n",
      "    perl-constant-1.33         |          pl526_1          10 KB  bioconda\n",
      "    perl-convert-binary-c-0.78 |  pl526h6bb024c_3         252 KB  bioconda\n",
      "    perl-convert-binhex-1.125  |          pl526_1          25 KB  bioconda\n",
      "    perl-crypt-rc4-2.02        |          pl526_1           8 KB  bioconda\n",
      "    perl-data-dumper-2.173     |          pl526_0          35 KB  bioconda\n",
      "    perl-data-optlist-0.110    |          pl526_2          10 KB  bioconda\n",
      "    perl-data-stag-0.14        |          pl526_1          78 KB  bioconda\n",
      "    perl-date-format-2.30      |          pl526_2          29 KB  bioconda\n",
      "    perl-db-file-1.852         |  pl526h14c3975_0          52 KB  bioconda\n",
      "    perl-dbd-sqlite-1.64       |  pl526h516909a_0         2.3 MB  bioconda\n",
      "    perl-dbi-1.642             |          pl526_0         549 KB  bioconda\n",
      "    perl-devel-globaldestruction-0.14|          pl526_0           7 KB  bioconda\n",
      "    perl-devel-overloadinfo-0.005|          pl526_0           8 KB  bioconda\n",
      "    perl-devel-stacktrace-2.04 |          pl526_0          16 KB  bioconda\n",
      "    perl-digest-hmac-1.03      |          pl526_3           9 KB  bioconda\n",
      "    perl-digest-md5-2.55       |          pl526_0          18 KB  bioconda\n",
      "    perl-digest-perl-md5-1.9   |          pl526_1          11 KB  bioconda\n",
      "    perl-digest-sha1-2.13      |  pl526h6bb024c_1          20 KB  bioconda\n",
      "    perl-dist-checkconflicts-0.11|          pl526_2          10 KB  bioconda\n",
      "    perl-dynaloader-1.25       |          pl526_1           3 KB  bioconda\n",
      "    perl-email-date-format-1.005|          pl526_2           8 KB  bioconda\n",
      "    perl-encode-2.88           |          pl526_1         2.1 MB  bioconda\n",
      "    perl-encode-locale-1.05    |          pl526_6          11 KB  bioconda\n",
      "    perl-error-0.17027         |          pl526_1          20 KB  bioconda\n",
      "    perl-eval-closure-0.14     |  pl526h6bb024c_4          11 KB  bioconda\n",
      "    perl-exception-class-1.44  |          pl526_0          19 KB  bioconda\n",
      "    perl-exporter-5.72         |          pl526_1          13 KB  bioconda\n",
      "    perl-exporter-tiny-1.002001|          pl526_0          23 KB  bioconda\n",
      "    perl-extutils-makemaker-7.36|          pl526_1         153 KB  bioconda\n",
      "    perl-file-listing-6.04     |          pl526_1           9 KB  bioconda\n",
      "    perl-file-path-2.16        |          pl526_0          17 KB  bioconda\n",
      "    perl-file-slurp-tiny-0.004 |          pl526_1           8 KB  bioconda\n",
      "    perl-file-sort-1.01        |          pl526_2          18 KB  bioconda\n",
      "    perl-file-temp-0.2304      |          pl526_2          26 KB  bioconda\n",
      "    perl-file-which-1.23       |          pl526_0          12 KB  bioconda\n",
      "    perl-font-afm-1.20         |          pl526_2          14 KB  bioconda\n",
      "    perl-font-ttf-1.06         |          pl526_0         200 KB  bioconda\n",
      "    perl-gd-2.68               |  pl526he941832_0         112 KB  bioconda\n",
      "    perl-getopt-long-2.50      |          pl526_1          27 KB  bioconda\n",
      "    perl-graph-0.9704          |          pl526_1          73 KB  bioconda\n",
      "    perl-graphviz-2.24         |  pl526h734ff71_0          36 KB  bioconda\n",
      "    perl-html-element-extended-1.18|          pl526_1          24 KB  bioconda\n",
      "    perl-html-entities-numbered-0.04|          pl526_1          11 KB  bioconda\n",
      "    perl-html-formatter-2.16   |          pl526_0          30 KB  bioconda\n",
      "    perl-html-parser-3.72      |  pl526h6bb024c_5          66 KB  bioconda\n",
      "    perl-html-tableextract-2.13|          pl526_2          23 KB  bioconda\n",
      "    perl-html-tagset-3.20      |          pl526_3          12 KB  bioconda\n",
      "    perl-html-tidy-1.60        |          pl526_0          20 KB  bioconda\n",
      "    perl-html-tree-5.07        |          pl526_1         129 KB  bioconda\n",
      "    perl-html-treebuilder-xpath-0.14|          pl526_1          12 KB  bioconda\n",
      "    perl-http-cookies-6.04     |          pl526_0          19 KB  bioconda\n",
      "    perl-http-daemon-6.01      |          pl526_1          14 KB  bioconda\n",
      "    perl-http-date-6.02        |          pl526_3          10 KB  bioconda\n",
      "    perl-http-message-6.18     |          pl526_0          51 KB  bioconda\n",
      "    perl-http-negotiate-6.01   |          pl526_3          13 KB  bioconda\n",
      "    perl-image-info-1.38       |          pl526_1          41 KB  bioconda\n",
      "    perl-image-size-3.300      |          pl526_2          26 KB  bioconda\n",
      "    perl-io-html-1.001         |          pl526_2          13 KB  bioconda\n",
      "    perl-io-sessiondata-1.03   |          pl526_1           8 KB  bioconda\n",
      "    perl-io-socket-ssl-2.066   |          pl526_0         151 KB  bioconda\n",
      "    perl-io-string-1.08        |          pl526_3          10 KB  bioconda\n",
      "    perl-io-stringy-2.111      |          pl526_1          35 KB  bioconda\n",
      "    perl-io-tty-1.12           |          pl526_1          30 KB  bioconda\n",
      "    perl-ipc-run-20180523.0    |          pl526_0          74 KB  bioconda\n",
      "    perl-ipc-sharelite-0.17    |  pl526h6bb024c_1          24 KB  bioconda\n",
      "    perl-jcode-2.07            |          pl526_2          24 KB  bioconda\n",
      "    perl-json-4.02             |          pl526_0          52 KB  bioconda\n",
      "    perl-json-xs-2.34          |  pl526h6bb024c_3          63 KB  bioconda\n",
      "    perl-lib-0.63              |          pl526_1           6 KB  bioconda\n",
      "    perl-libwww-perl-6.39      |          pl526_0          94 KB  bioconda\n",
      "    perl-libxml-perl-0.08      |          pl526_2          38 KB  bioconda\n",
      "    perl-list-moreutils-0.428  |          pl526_1          28 KB  bioconda\n",
      "    perl-list-moreutils-xs-0.428|          pl526_0          43 KB  bioconda\n",
      "    perl-lwp-mediatypes-6.04   |          pl526_0          22 KB  bioconda\n",
      "    perl-lwp-protocol-https-6.07|          pl526_4           9 KB  bioconda\n",
      "    perl-lwp-simple-6.15       |  pl526h470a237_4         348 KB  bioconda\n",
      "    perl-mailtools-2.21        |          pl526_0          48 KB  bioconda\n",
      "    perl-math-cdf-0.1          |  pl526h14c3975_5          69 KB  bioconda\n",
      "    perl-math-derivative-1.01  |          pl526_0           9 KB  bioconda\n",
      "    perl-math-random-0.72      |  pl526h14c3975_2          55 KB  bioconda\n",
      "    perl-math-spline-0.02      |          pl526_2           8 KB  bioconda\n",
      "    perl-mime-base64-3.15      |          pl526_1          15 KB  bioconda\n",
      "    perl-mime-lite-3.030       |          pl526_1          51 KB  bioconda\n",
      "    perl-mime-tools-5.508      |          pl526_1         113 KB  bioconda\n",
      "    perl-mime-types-2.17       |          pl526_0          56 KB  bioconda\n",
      "    perl-mldbm-2.05            |          pl526_1          15 KB  bioconda\n",
      "    perl-module-implementation-0.09|          pl526_2           9 KB  bioconda\n",
      "    perl-module-runtime-0.016  |          pl526_1          15 KB  bioconda\n",
      "    perl-module-runtime-conflicts-0.003|          pl526_0           7 KB  bioconda\n",
      "    perl-moo-2.003004          |          pl526_0          38 KB  bioconda\n",
      "    perl-moose-2.2011          |  pl526hf484d3e_1         444 KB  bioconda\n",
      "    perl-mozilla-ca-20180117   |          pl526_1         140 KB  bioconda\n",
      "    perl-mro-compat-0.13       |          pl526_0          10 KB  bioconda\n",
      "    perl-net-http-6.19         |          pl526_0          19 KB  bioconda\n",
      "    perl-net-ssleay-1.88       |  pl526h90d6eec_0         289 KB  bioconda\n",
      "    perl-ntlm-1.09             |          pl526_4          15 KB  bioconda\n",
      "    perl-ole-storage_lite-0.19 |          pl526_3          18 KB  bioconda\n",
      "    perl-package-deprecationmanager-0.17|          pl526_0          10 KB  bioconda\n",
      "    perl-package-stash-0.38    |  pl526hf484d3e_1          66 KB  bioconda\n",
      "    perl-package-stash-xs-0.28 |  pl526hf484d3e_1          21 KB  bioconda\n",
      "    perl-params-util-1.07      |  pl526h6bb024c_4          14 KB  bioconda\n",
      "    perl-parent-0.236          |          pl526_1           5 KB  bioconda\n",
      "    perl-parse-recdescent-1.967015|          pl526_0          71 KB  bioconda\n",
      "    perl-pathtools-3.75        |  pl526h14c3975_1          41 KB  bioconda\n",
      "    perl-pdf-api2-2.035        |          pl526_0         2.3 MB  bioconda\n",
      "    perl-pod-escapes-1.07      |          pl526_1          10 KB  bioconda\n",
      "    perl-pod-usage-1.69        |          pl526_1          13 KB  bioconda\n",
      "    perl-postscript-0.06       |          pl526_2          22 KB  bioconda\n",
      "    perl-role-tiny-2.000008    |          pl526_0          15 KB  bioconda\n",
      "    perl-scalar-list-utils-1.52|  pl526h516909a_0          38 KB  bioconda\n",
      "    perl-set-scalar-1.29       |          pl526_2          19 KB  bioconda\n",
      "    perl-soap-lite-1.19        |          pl526_1         117 KB  bioconda\n",
      "    perl-socket-2.027          |          pl526_1          31 KB  bioconda\n",
      "    perl-sort-naturally-1.03   |          pl526_2          14 KB  bioconda\n",
      "    perl-spreadsheet-parseexcel-0.65|          pl526_2          63 KB  bioconda\n",
      "    perl-spreadsheet-writeexcel-2.40|          pl526_2         259 KB  bioconda\n",
      "    perl-statistics-descriptive-3.0702|          pl526_0          32 KB  bioconda\n",
      "    perl-storable-3.15         |  pl526h14c3975_0          63 KB  bioconda\n",
      "    perl-sub-exporter-0.987    |          pl526_2          30 KB  bioconda\n",
      "    perl-sub-exporter-progressive-0.001013|          pl526_0           8 KB  bioconda\n",
      "    perl-sub-identify-0.14     |  pl526h14c3975_0          12 KB  bioconda\n",
      "    perl-sub-install-0.928     |          pl526_2          10 KB  bioconda\n",
      "    perl-sub-name-0.21         |          pl526_1          13 KB  bioconda\n",
      "    perl-sub-quote-2.006003    |          pl526_1          18 KB  bioconda\n",
      "    perl-sub-uplevel-0.2800    |  pl526h14c3975_2          13 KB  bioconda\n",
      "    perl-svg-2.84              |          pl526_0          37 KB  bioconda\n",
      "    perl-svg-graph-0.02        |          pl526_3          26 KB  bioconda\n",
      "    perl-task-weaken-1.06      |          pl526_0           8 KB  bioconda\n",
      "    perl-template-toolkit-2.26 |          pl526_1         319 KB  bioconda\n",
      "    perl-test-1.26             |          pl526_1          14 KB  bioconda\n",
      "    perl-test-deep-1.128       |          pl526_1          36 KB  bioconda\n",
      "    perl-test-differences-0.67 |          pl526_0          15 KB  bioconda\n",
      "    perl-test-exception-0.43   |          pl526_2          12 KB  bioconda\n",
      "    perl-test-harness-3.42     |          pl526_0          82 KB  bioconda\n",
      "    perl-test-leaktrace-0.16   |  pl526h14c3975_2          23 KB  bioconda\n",
      "    perl-test-most-0.35        |          pl526_0          18 KB  bioconda\n",
      "    perl-test-requiresinternet-0.05|          pl526_0           7 KB  bioconda\n",
      "    perl-test-warn-0.36        |          pl526_1          13 KB  bioconda\n",
      "    perl-text-diff-1.45        |          pl526_0          20 KB  bioconda\n",
      "    perl-tie-ixhash-1.23       |          pl526_2          11 KB  bioconda\n",
      "    perl-time-hires-1.9760     |  pl526h14c3975_1          25 KB  bioconda\n",
      "    perl-time-local-1.28       |          pl526_1          10 KB  bioconda\n",
      "    perl-timedate-2.30         |          pl526_1          29 KB  bioconda\n",
      "    perl-tree-dag_node-1.31    |          pl526_0          40 KB  bioconda\n",
      "    perl-try-tiny-0.30         |          pl526_1          17 KB  bioconda\n",
      "    perl-type-tiny-1.004004    |          pl526_0         119 KB  bioconda\n",
      "    perl-types-serialiser-1.0  |          pl526_2          10 KB  bioconda\n",
      "    perl-unicode-map-0.112     |  pl526h6bb024c_3         254 KB  bioconda\n",
      "    perl-uri-1.76              |          pl526_0          55 KB  bioconda\n",
      "    perl-www-robotrules-6.02   |          pl526_3          12 KB  bioconda\n",
      "    perl-xml-dom-1.46          |          pl526_0          56 KB  bioconda\n",
      "    perl-xml-dom-xpath-0.14    |          pl526_1           9 KB  bioconda\n",
      "    perl-xml-filter-buffertext-1.01|          pl526_2           7 KB  bioconda\n",
      "    perl-xml-libxml-2.0132     |  pl526h7ec2d77_1         260 KB  bioconda\n",
      "    perl-xml-libxslt-1.94      |          pl526_1          40 KB  bioconda\n",
      "    perl-xml-namespacesupport-1.12|          pl526_0          11 KB  bioconda\n",
      "    perl-xml-parser-2.44       |                4         151 KB  bioconda\n",
      "    perl-xml-regexp-0.04       |          pl526_2           8 KB  bioconda\n",
      "    perl-xml-sax-1.02          |          pl526_0          36 KB  bioconda\n",
      "    perl-xml-sax-base-1.09     |          pl526_0          19 KB  bioconda\n",
      "    perl-xml-sax-expat-0.51    |          pl526_3          10 KB  bioconda\n",
      "    perl-xml-sax-writer-0.57   |          pl526_0          16 KB  bioconda\n",
      "    perl-xml-simple-2.25       |          pl526_1          48 KB  bioconda\n",
      "    perl-xml-twig-3.52         |          pl526_2         144 KB  bioconda\n",
      "    perl-xml-writer-0.625      |          pl526_2          18 KB  bioconda\n",
      "    perl-xml-xpath-1.44        |          pl526_0          42 KB  bioconda\n",
      "    perl-xml-xpathengine-0.14  |          pl526_2          26 KB  bioconda\n",
      "    perl-xsloader-0.24         |          pl526_0           8 KB  bioconda\n",
      "    samclip-0.4.0              |       hdfd78af_1           7 KB  bioconda\n",
      "    samtools-1.14              |       hb421002_0         411 KB  bioconda\n",
      "    snippy-4.5.0               |                0        18.9 MB  bioconda\n",
      "    snp-sites-2.5.1            |       h5bf99c6_1          41 KB  bioconda\n",
      "    snpeff-5.0                 |       hdfd78af_1        23.2 MB  bioconda\n",
      "    t_coffee-11.0.8            |   py37hea885bf_8        33.2 MB  bioconda\n",
      "    tidyp-1.04                 |       h779adbc_3         300 KB  bioconda\n",
      "    vcflib-1.0.0_rc3           |   py37hc088bd4_0        26.6 MB  bioconda\n",
      "    vt-2015.11.10              |       he941832_3         1.5 MB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:       312.8 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  any2fasta          bioconda/noarch::any2fasta-0.4.2-hdfd78af_3\n",
      "  atk-1.0            conda-forge/linux-64::atk-1.0-2.36.0-h3371d22_4\n",
      "  bcftools           bioconda/linux-64::bcftools-1.14-h88f3f91_0\n",
      "  bedtools           bioconda/linux-64::bedtools-2.30.0-h7d7f7ad_2\n",
      "  bwa                bioconda/linux-64::bwa-0.7.17-h5bf99c6_8\n",
      "  clustalw           bioconda/linux-64::clustalw-2.1-h7d875b9_6\n",
      "  curl               conda-forge/linux-64::curl-7.80.0-h2574ce0_0\n",
      "  font-ttf-dejavu-s~ conda-forge/noarch::font-ttf-dejavu-sans-mono-2.37-hab24e00_0\n",
      "  font-ttf-inconsol~ conda-forge/noarch::font-ttf-inconsolata-3.000-h77eed37_0\n",
      "  font-ttf-source-c~ conda-forge/noarch::font-ttf-source-code-pro-2.038-h77eed37_0\n",
      "  font-ttf-ubuntu    conda-forge/noarch::font-ttf-ubuntu-0.83-hab24e00_0\n",
      "  fonts-conda-ecosy~ conda-forge/noarch::fonts-conda-ecosystem-1-0\n",
      "  fonts-conda-forge  conda-forge/noarch::fonts-conda-forge-1-0\n",
      "  freebayes          bioconda/linux-64::freebayes-1.3.2-py37h26878c9_2\n",
      "  fribidi            conda-forge/linux-64::fribidi-1.0.10-h36c2ea0_0\n",
      "  gdk-pixbuf         conda-forge/linux-64::gdk-pixbuf-2.42.6-h04a7f16_0\n",
      "  giflib             conda-forge/linux-64::giflib-5.2.1-h36c2ea0_2\n",
      "  graphviz           conda-forge/linux-64::graphviz-2.50.0-h85b4f2f_1\n",
      "  gsl                conda-forge/linux-64::gsl-2.6-he838d99_2\n",
      "  gtk2               conda-forge/linux-64::gtk2-2.24.33-h539f30e_1\n",
      "  gts                conda-forge/linux-64::gts-0.7.6-h64030ff_2\n",
      "  htslib             bioconda/linux-64::htslib-1.14-h9093b5e_0\n",
      "  k8                 bioconda/linux-64::k8-0.2.5-h9a82719_1\n",
      "  libdb              bioconda/linux-64::libdb-6.1.26-0\n",
      "  libgd              conda-forge/linux-64::libgd-2.3.3-h6ad9fb6_0\n",
      "  librsvg            conda-forge/linux-64::librsvg-2.52.4-hc3c00ef_0\n",
      "  libtool            conda-forge/linux-64::libtool-2.4.6-h9c3ff4c_1008\n",
      "  libwebp            conda-forge/linux-64::libwebp-1.2.1-h3452ae3_0\n",
      "  libxslt            conda-forge/linux-64::libxslt-1.1.33-h15afd5d_2\n",
      "  minimap2           bioconda/linux-64::minimap2-2.23-h5bf99c6_0\n",
      "  openjdk            conda-forge/linux-64::openjdk-8.0.312-h7f98852_0\n",
      "  paml               bioconda/linux-64::paml-4.9-h779adbc_6\n",
      "  pango              conda-forge/linux-64::pango-1.48.10-h54213e6_2\n",
      "  parallel           conda-forge/linux-64::parallel-20211022-ha770c72_0\n",
      "  perl-aceperl       bioconda/linux-64::perl-aceperl-1.92-pl526_2\n",
      "  perl-algorithm-di~ bioconda/linux-64::perl-algorithm-diff-1.1903-pl526_2\n",
      "  perl-algorithm-mu~ bioconda/linux-64::perl-algorithm-munkres-0.08-pl526_1\n",
      "  perl-apache-test   bioconda/linux-64::perl-apache-test-1.40-pl526_1\n",
      "  perl-app-cpanminus bioconda/linux-64::perl-app-cpanminus-1.7044-pl526_1\n",
      "  perl-appconfig     bioconda/linux-64::perl-appconfig-1.71-pl526_1\n",
      "  perl-array-compare bioconda/linux-64::perl-array-compare-3.0.1-pl526_1\n",
      "  perl-autoloader    bioconda/linux-64::perl-autoloader-5.74-pl526_2\n",
      "  perl-base          bioconda/linux-64::perl-base-2.23-pl526_1\n",
      "  perl-bio-asn1-ent~ bioconda/noarch::perl-bio-asn1-entrezgene-1.73-pl5262hdfd78af_2\n",
      "  perl-bio-coordina~ bioconda/noarch::perl-bio-coordinate-1.007001-pl5262hdfd78af_2\n",
      "  perl-bio-featureio bioconda/noarch::perl-bio-featureio-1.6.905-pl5262hdfd78af_3\n",
      "  perl-bio-phylo     bioconda/noarch::perl-bio-phylo-0.58-pl5262hdfd78af_3\n",
      "  perl-bio-samtools  bioconda/linux-64::perl-bio-samtools-1.43-pl526h1341992_1\n",
      "  perl-bio-tools-ph~ bioconda/noarch::perl-bio-tools-phylo-paml-1.7.3-pl5262hdfd78af_2\n",
      "  perl-bio-tools-ru~ bioconda/noarch::perl-bio-tools-run-alignment-clustalw-1.7.4-pl5262hdfd78af_2\n",
      "  perl-bio-tools-ru~ bioconda/noarch::perl-bio-tools-run-alignment-tcoffee-1.7.4-pl5262hdfd78af_3\n",
      "  perl-bioperl-core  bioconda/noarch::perl-bioperl-core-1.007002-pl5262hdfd78af_3\n",
      "  perl-bioperl-run   bioconda/noarch::perl-bioperl-run-1.007002-pl5262hdfd78af_5\n",
      "  perl-business-isbn bioconda/linux-64::perl-business-isbn-3.004-pl526_0\n",
      "  perl-business-isb~ bioconda/linux-64::perl-business-isbn-data-20140910.003-pl526_0\n",
      "  perl-cache-cache   bioconda/linux-64::perl-cache-cache-1.08-pl526_0\n",
      "  perl-capture-tiny  bioconda/linux-64::perl-capture-tiny-0.48-pl526_0\n",
      "  perl-carp          bioconda/linux-64::perl-carp-1.38-pl526_3\n",
      "  perl-cgi           bioconda/linux-64::perl-cgi-4.44-pl526h14c3975_1\n",
      "  perl-class-data-i~ bioconda/linux-64::perl-class-data-inheritable-0.08-pl526_1\n",
      "  perl-class-inspec~ bioconda/linux-64::perl-class-inspector-1.34-pl526_0\n",
      "  perl-class-load    bioconda/linux-64::perl-class-load-0.25-pl526_0\n",
      "  perl-class-load-xs bioconda/linux-64::perl-class-load-xs-0.10-pl526h6bb024c_2\n",
      "  perl-class-method~ bioconda/linux-64::perl-class-method-modifiers-2.12-pl526_0\n",
      "  perl-clone         bioconda/linux-64::perl-clone-0.42-pl526h516909a_0\n",
      "  perl-common-sense  bioconda/linux-64::perl-common-sense-3.74-pl526_2\n",
      "  perl-compress-raw~ bioconda/linux-64::perl-compress-raw-zlib-2.087-pl526hc9558a2_0\n",
      "  perl-constant      bioconda/linux-64::perl-constant-1.33-pl526_1\n",
      "  perl-convert-bina~ bioconda/linux-64::perl-convert-binary-c-0.78-pl526h6bb024c_3\n",
      "  perl-convert-binh~ bioconda/linux-64::perl-convert-binhex-1.125-pl526_1\n",
      "  perl-crypt-rc4     bioconda/linux-64::perl-crypt-rc4-2.02-pl526_1\n",
      "  perl-data-dumper   bioconda/linux-64::perl-data-dumper-2.173-pl526_0\n",
      "  perl-data-optlist  bioconda/linux-64::perl-data-optlist-0.110-pl526_2\n",
      "  perl-data-stag     bioconda/linux-64::perl-data-stag-0.14-pl526_1\n",
      "  perl-date-format   bioconda/linux-64::perl-date-format-2.30-pl526_2\n",
      "  perl-db-file       bioconda/linux-64::perl-db-file-1.852-pl526h14c3975_0\n",
      "  perl-dbd-sqlite    bioconda/linux-64::perl-dbd-sqlite-1.64-pl526h516909a_0\n",
      "  perl-dbi           bioconda/linux-64::perl-dbi-1.642-pl526_0\n",
      "  perl-devel-global~ bioconda/linux-64::perl-devel-globaldestruction-0.14-pl526_0\n",
      "  perl-devel-overlo~ bioconda/linux-64::perl-devel-overloadinfo-0.005-pl526_0\n",
      "  perl-devel-stackt~ bioconda/noarch::perl-devel-stacktrace-2.04-pl526_0\n",
      "  perl-digest-hmac   bioconda/linux-64::perl-digest-hmac-1.03-pl526_3\n",
      "  perl-digest-md5    bioconda/linux-64::perl-digest-md5-2.55-pl526_0\n",
      "  perl-digest-perl-~ bioconda/linux-64::perl-digest-perl-md5-1.9-pl526_1\n",
      "  perl-digest-sha1   bioconda/linux-64::perl-digest-sha1-2.13-pl526h6bb024c_1\n",
      "  perl-dist-checkco~ bioconda/linux-64::perl-dist-checkconflicts-0.11-pl526_2\n",
      "  perl-dynaloader    bioconda/linux-64::perl-dynaloader-1.25-pl526_1\n",
      "  perl-email-date-f~ bioconda/linux-64::perl-email-date-format-1.005-pl526_2\n",
      "  perl-encode        bioconda/linux-64::perl-encode-2.88-pl526_1\n",
      "  perl-encode-locale bioconda/linux-64::perl-encode-locale-1.05-pl526_6\n",
      "  perl-error         bioconda/linux-64::perl-error-0.17027-pl526_1\n",
      "  perl-eval-closure  bioconda/linux-64::perl-eval-closure-0.14-pl526h6bb024c_4\n",
      "  perl-exception-cl~ bioconda/linux-64::perl-exception-class-1.44-pl526_0\n",
      "  perl-exporter      bioconda/linux-64::perl-exporter-5.72-pl526_1\n",
      "  perl-exporter-tiny bioconda/linux-64::perl-exporter-tiny-1.002001-pl526_0\n",
      "  perl-extutils-mak~ bioconda/linux-64::perl-extutils-makemaker-7.36-pl526_1\n",
      "  perl-file-listing  bioconda/linux-64::perl-file-listing-6.04-pl526_1\n",
      "  perl-file-path     bioconda/linux-64::perl-file-path-2.16-pl526_0\n",
      "  perl-file-slurp-t~ bioconda/linux-64::perl-file-slurp-tiny-0.004-pl526_1\n",
      "  perl-file-sort     bioconda/linux-64::perl-file-sort-1.01-pl526_2\n",
      "  perl-file-temp     bioconda/linux-64::perl-file-temp-0.2304-pl526_2\n",
      "  perl-file-which    bioconda/linux-64::perl-file-which-1.23-pl526_0\n",
      "  perl-font-afm      bioconda/linux-64::perl-font-afm-1.20-pl526_2\n",
      "  perl-font-ttf      bioconda/linux-64::perl-font-ttf-1.06-pl526_0\n",
      "  perl-gd            bioconda/linux-64::perl-gd-2.68-pl526he941832_0\n",
      "  perl-getopt-long   bioconda/linux-64::perl-getopt-long-2.50-pl526_1\n",
      "  perl-graph         bioconda/linux-64::perl-graph-0.9704-pl526_1\n",
      "  perl-graphviz      bioconda/linux-64::perl-graphviz-2.24-pl526h734ff71_0\n",
      "  perl-html-element~ bioconda/linux-64::perl-html-element-extended-1.18-pl526_1\n",
      "  perl-html-entitie~ bioconda/linux-64::perl-html-entities-numbered-0.04-pl526_1\n",
      "  perl-html-formatt~ bioconda/linux-64::perl-html-formatter-2.16-pl526_0\n",
      "  perl-html-parser   bioconda/linux-64::perl-html-parser-3.72-pl526h6bb024c_5\n",
      "  perl-html-tableex~ bioconda/linux-64::perl-html-tableextract-2.13-pl526_2\n",
      "  perl-html-tagset   bioconda/linux-64::perl-html-tagset-3.20-pl526_3\n",
      "  perl-html-tidy     bioconda/linux-64::perl-html-tidy-1.60-pl526_0\n",
      "  perl-html-tree     bioconda/linux-64::perl-html-tree-5.07-pl526_1\n",
      "  perl-html-treebui~ bioconda/linux-64::perl-html-treebuilder-xpath-0.14-pl526_1\n",
      "  perl-http-cookies  bioconda/linux-64::perl-http-cookies-6.04-pl526_0\n",
      "  perl-http-daemon   bioconda/linux-64::perl-http-daemon-6.01-pl526_1\n",
      "  perl-http-date     bioconda/linux-64::perl-http-date-6.02-pl526_3\n",
      "  perl-http-message  bioconda/linux-64::perl-http-message-6.18-pl526_0\n",
      "  perl-http-negotia~ bioconda/linux-64::perl-http-negotiate-6.01-pl526_3\n",
      "  perl-image-info    bioconda/linux-64::perl-image-info-1.38-pl526_1\n",
      "  perl-image-size    bioconda/linux-64::perl-image-size-3.300-pl526_2\n",
      "  perl-io-html       bioconda/linux-64::perl-io-html-1.001-pl526_2\n",
      "  perl-io-sessionda~ bioconda/linux-64::perl-io-sessiondata-1.03-pl526_1\n",
      "  perl-io-socket-ssl bioconda/linux-64::perl-io-socket-ssl-2.066-pl526_0\n",
      "  perl-io-string     bioconda/linux-64::perl-io-string-1.08-pl526_3\n",
      "  perl-io-stringy    bioconda/linux-64::perl-io-stringy-2.111-pl526_1\n",
      "  perl-io-tty        bioconda/linux-64::perl-io-tty-1.12-pl526_1\n",
      "  perl-ipc-run       bioconda/linux-64::perl-ipc-run-20180523.0-pl526_0\n",
      "  perl-ipc-sharelite bioconda/linux-64::perl-ipc-sharelite-0.17-pl526h6bb024c_1\n",
      "  perl-jcode         bioconda/linux-64::perl-jcode-2.07-pl526_2\n",
      "  perl-json          bioconda/linux-64::perl-json-4.02-pl526_0\n",
      "  perl-json-xs       bioconda/linux-64::perl-json-xs-2.34-pl526h6bb024c_3\n",
      "  perl-lib           bioconda/linux-64::perl-lib-0.63-pl526_1\n",
      "  perl-libwww-perl   bioconda/noarch::perl-libwww-perl-6.39-pl526_0\n",
      "  perl-libxml-perl   bioconda/linux-64::perl-libxml-perl-0.08-pl526_2\n",
      "  perl-list-moreuti~ bioconda/linux-64::perl-list-moreutils-0.428-pl526_1\n",
      "  perl-list-moreuti~ bioconda/linux-64::perl-list-moreutils-xs-0.428-pl526_0\n",
      "  perl-lwp-mediatyp~ bioconda/linux-64::perl-lwp-mediatypes-6.04-pl526_0\n",
      "  perl-lwp-protocol~ bioconda/linux-64::perl-lwp-protocol-https-6.07-pl526_4\n",
      "  perl-lwp-simple    bioconda/linux-64::perl-lwp-simple-6.15-pl526h470a237_4\n",
      "  perl-mailtools     bioconda/noarch::perl-mailtools-2.21-pl526_0\n",
      "  perl-math-cdf      bioconda/linux-64::perl-math-cdf-0.1-pl526h14c3975_5\n",
      "  perl-math-derivat~ bioconda/linux-64::perl-math-derivative-1.01-pl526_0\n",
      "  perl-math-random   bioconda/linux-64::perl-math-random-0.72-pl526h14c3975_2\n",
      "  perl-math-spline   bioconda/linux-64::perl-math-spline-0.02-pl526_2\n",
      "  perl-mime-base64   bioconda/linux-64::perl-mime-base64-3.15-pl526_1\n",
      "  perl-mime-lite     bioconda/linux-64::perl-mime-lite-3.030-pl526_1\n",
      "  perl-mime-tools    bioconda/linux-64::perl-mime-tools-5.508-pl526_1\n",
      "  perl-mime-types    bioconda/linux-64::perl-mime-types-2.17-pl526_0\n",
      "  perl-mldbm         bioconda/linux-64::perl-mldbm-2.05-pl526_1\n",
      "  perl-module-imple~ bioconda/linux-64::perl-module-implementation-0.09-pl526_2\n",
      "  perl-module-runti~ bioconda/linux-64::perl-module-runtime-0.016-pl526_1\n",
      "  perl-module-runti~ bioconda/linux-64::perl-module-runtime-conflicts-0.003-pl526_0\n",
      "  perl-moo           bioconda/linux-64::perl-moo-2.003004-pl526_0\n",
      "  perl-moose         bioconda/linux-64::perl-moose-2.2011-pl526hf484d3e_1\n",
      "  perl-mozilla-ca    bioconda/linux-64::perl-mozilla-ca-20180117-pl526_1\n",
      "  perl-mro-compat    bioconda/linux-64::perl-mro-compat-0.13-pl526_0\n",
      "  perl-net-http      bioconda/noarch::perl-net-http-6.19-pl526_0\n",
      "  perl-net-ssleay    bioconda/linux-64::perl-net-ssleay-1.88-pl526h90d6eec_0\n",
      "  perl-ntlm          bioconda/linux-64::perl-ntlm-1.09-pl526_4\n",
      "  perl-ole-storage_~ bioconda/linux-64::perl-ole-storage_lite-0.19-pl526_3\n",
      "  perl-package-depr~ bioconda/linux-64::perl-package-deprecationmanager-0.17-pl526_0\n",
      "  perl-package-stash bioconda/linux-64::perl-package-stash-0.38-pl526hf484d3e_1\n",
      "  perl-package-stas~ bioconda/linux-64::perl-package-stash-xs-0.28-pl526hf484d3e_1\n",
      "  perl-params-util   bioconda/linux-64::perl-params-util-1.07-pl526h6bb024c_4\n",
      "  perl-parent        bioconda/linux-64::perl-parent-0.236-pl526_1\n",
      "  perl-parse-recdes~ bioconda/linux-64::perl-parse-recdescent-1.967015-pl526_0\n",
      "  perl-pathtools     bioconda/linux-64::perl-pathtools-3.75-pl526h14c3975_1\n",
      "  perl-pdf-api2      bioconda/noarch::perl-pdf-api2-2.035-pl526_0\n",
      "  perl-pod-escapes   bioconda/linux-64::perl-pod-escapes-1.07-pl526_1\n",
      "  perl-pod-usage     bioconda/linux-64::perl-pod-usage-1.69-pl526_1\n",
      "  perl-postscript    bioconda/linux-64::perl-postscript-0.06-pl526_2\n",
      "  perl-role-tiny     bioconda/noarch::perl-role-tiny-2.000008-pl526_0\n",
      "  perl-scalar-list-~ bioconda/linux-64::perl-scalar-list-utils-1.52-pl526h516909a_0\n",
      "  perl-set-scalar    bioconda/linux-64::perl-set-scalar-1.29-pl526_2\n",
      "  perl-soap-lite     bioconda/linux-64::perl-soap-lite-1.19-pl526_1\n",
      "  perl-socket        bioconda/linux-64::perl-socket-2.027-pl526_1\n",
      "  perl-sort-natural~ bioconda/linux-64::perl-sort-naturally-1.03-pl526_2\n",
      "  perl-spreadsheet-~ bioconda/linux-64::perl-spreadsheet-parseexcel-0.65-pl526_2\n",
      "  perl-spreadsheet-~ bioconda/linux-64::perl-spreadsheet-writeexcel-2.40-pl526_2\n",
      "  perl-statistics-d~ bioconda/linux-64::perl-statistics-descriptive-3.0702-pl526_0\n",
      "  perl-storable      bioconda/linux-64::perl-storable-3.15-pl526h14c3975_0\n",
      "  perl-sub-exporter  bioconda/linux-64::perl-sub-exporter-0.987-pl526_2\n",
      "  perl-sub-exporter~ bioconda/linux-64::perl-sub-exporter-progressive-0.001013-pl526_0\n",
      "  perl-sub-identify  bioconda/linux-64::perl-sub-identify-0.14-pl526h14c3975_0\n",
      "  perl-sub-install   bioconda/linux-64::perl-sub-install-0.928-pl526_2\n",
      "  perl-sub-name      bioconda/linux-64::perl-sub-name-0.21-pl526_1\n",
      "  perl-sub-quote     bioconda/linux-64::perl-sub-quote-2.006003-pl526_1\n",
      "  perl-sub-uplevel   bioconda/linux-64::perl-sub-uplevel-0.2800-pl526h14c3975_2\n",
      "  perl-svg           bioconda/linux-64::perl-svg-2.84-pl526_0\n",
      "  perl-svg-graph     bioconda/linux-64::perl-svg-graph-0.02-pl526_3\n",
      "  perl-task-weaken   bioconda/linux-64::perl-task-weaken-1.06-pl526_0\n",
      "  perl-template-too~ bioconda/linux-64::perl-template-toolkit-2.26-pl526_1\n",
      "  perl-test          bioconda/linux-64::perl-test-1.26-pl526_1\n",
      "  perl-test-deep     bioconda/linux-64::perl-test-deep-1.128-pl526_1\n",
      "  perl-test-differe~ bioconda/noarch::perl-test-differences-0.67-pl526_0\n",
      "  perl-test-excepti~ bioconda/linux-64::perl-test-exception-0.43-pl526_2\n",
      "  perl-test-harness  bioconda/linux-64::perl-test-harness-3.42-pl526_0\n",
      "  perl-test-leaktra~ bioconda/linux-64::perl-test-leaktrace-0.16-pl526h14c3975_2\n",
      "  perl-test-most     bioconda/linux-64::perl-test-most-0.35-pl526_0\n",
      "  perl-test-require~ bioconda/linux-64::perl-test-requiresinternet-0.05-pl526_0\n",
      "  perl-test-warn     bioconda/linux-64::perl-test-warn-0.36-pl526_1\n",
      "  perl-text-diff     bioconda/linux-64::perl-text-diff-1.45-pl526_0\n",
      "  perl-tie-ixhash    bioconda/linux-64::perl-tie-ixhash-1.23-pl526_2\n",
      "  perl-time-hires    bioconda/linux-64::perl-time-hires-1.9760-pl526h14c3975_1\n",
      "  perl-time-local    bioconda/linux-64::perl-time-local-1.28-pl526_1\n",
      "  perl-timedate      bioconda/linux-64::perl-timedate-2.30-pl526_1\n",
      "  perl-tree-dag_node bioconda/linux-64::perl-tree-dag_node-1.31-pl526_0\n",
      "  perl-try-tiny      bioconda/linux-64::perl-try-tiny-0.30-pl526_1\n",
      "  perl-type-tiny     bioconda/linux-64::perl-type-tiny-1.004004-pl526_0\n",
      "  perl-types-serial~ bioconda/linux-64::perl-types-serialiser-1.0-pl526_2\n",
      "  perl-unicode-map   bioconda/linux-64::perl-unicode-map-0.112-pl526h6bb024c_3\n",
      "  perl-uri           bioconda/linux-64::perl-uri-1.76-pl526_0\n",
      "  perl-www-robotrul~ bioconda/linux-64::perl-www-robotrules-6.02-pl526_3\n",
      "  perl-xml-dom       bioconda/linux-64::perl-xml-dom-1.46-pl526_0\n",
      "  perl-xml-dom-xpath bioconda/linux-64::perl-xml-dom-xpath-0.14-pl526_1\n",
      "  perl-xml-filter-b~ bioconda/linux-64::perl-xml-filter-buffertext-1.01-pl526_2\n",
      "  perl-xml-libxml    bioconda/linux-64::perl-xml-libxml-2.0132-pl526h7ec2d77_1\n",
      "  perl-xml-libxslt   bioconda/linux-64::perl-xml-libxslt-1.94-pl526_1\n",
      "  perl-xml-namespac~ bioconda/linux-64::perl-xml-namespacesupport-1.12-pl526_0\n",
      "  perl-xml-parser    bioconda/linux-64::perl-xml-parser-2.44-4\n",
      "  perl-xml-regexp    bioconda/linux-64::perl-xml-regexp-0.04-pl526_2\n",
      "  perl-xml-sax       bioconda/noarch::perl-xml-sax-1.02-pl526_0\n",
      "  perl-xml-sax-base  bioconda/linux-64::perl-xml-sax-base-1.09-pl526_0\n",
      "  perl-xml-sax-expat bioconda/linux-64::perl-xml-sax-expat-0.51-pl526_3\n",
      "  perl-xml-sax-writ~ bioconda/linux-64::perl-xml-sax-writer-0.57-pl526_0\n",
      "  perl-xml-simple    bioconda/linux-64::perl-xml-simple-2.25-pl526_1\n",
      "  perl-xml-twig      bioconda/linux-64::perl-xml-twig-3.52-pl526_2\n",
      "  perl-xml-writer    bioconda/linux-64::perl-xml-writer-0.625-pl526_2\n",
      "  perl-xml-xpath     bioconda/linux-64::perl-xml-xpath-1.44-pl526_0\n",
      "  perl-xml-xpatheng~ bioconda/linux-64::perl-xml-xpathengine-0.14-pl526_2\n",
      "  perl-xsloader      bioconda/linux-64::perl-xsloader-0.24-pl526_0\n",
      "  samclip            bioconda/noarch::samclip-0.4.0-hdfd78af_1\n",
      "  samtools           bioconda/linux-64::samtools-1.14-hb421002_0\n",
      "  snippy             bioconda/noarch::snippy-4.5.0-0\n",
      "  snp-sites          bioconda/linux-64::snp-sites-2.5.1-h5bf99c6_1\n",
      "  snpeff             bioconda/noarch::snpeff-5.0-hdfd78af_1\n",
      "  t_coffee           bioconda/linux-64::t_coffee-11.0.8-py37hea885bf_8\n",
      "  tidyp              bioconda/linux-64::tidyp-1.04-h779adbc_3\n",
      "  vcflib             bioconda/linux-64::vcflib-1.0.0_rc3-py37hc088bd4_0\n",
      "  vt                 bioconda/linux-64::vt-2015.11.10-he941832_3\n",
      "\n",
      "The following packages will be UPDATED:\n",
      "\n",
      "  ca-certificates    anaconda::ca-certificates-2020.10.14-0 --> conda-forge::ca-certificates-2021.10.8-ha878542_0\n",
      "  certifi                anaconda::certifi-2020.6.20-py37_0 --> conda-forge::certifi-2021.10.8-py37h89c1867_1\n",
      "  perl-bioperl       bioconda/linux-64::perl-bioperl-1.6.9~ --> bioconda/noarch::perl-bioperl-1.7.2-pl526_11\n",
      "\n",
      "The following packages will be DOWNGRADED:\n",
      "\n",
      "  lerc                                       3.0-h9c3ff4c_0 --> 2.2.1-h9c3ff4c_0\n",
      "  libdeflate                                 1.8-h7f98852_0 --> 1.7-h7f98852_5\n",
      "  libtiff                                  4.3.0-h6f004c6_2 --> 4.3.0-hf544144_1\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "fonts-conda-forge-1  | 4 KB      | ##################################### | 100% \n",
      "perl-class-method-mo | 13 KB     | ##################################### | 100% \n",
      "perl-io-html-1.001   | 13 KB     | ##################################### | 100% \n",
      "perl-mozilla-ca-2018 | 140 KB    | ##################################### | 100% \n",
      "htslib-1.14          | 2.2 MB    | ##################################### | 100% \n",
      "libgd-2.3.3          | 279 KB    | ##################################### | 100% \n",
      "graphviz-2.50.0      | 2.9 MB    | ##################################### | 100% \n",
      "perl-http-daemon-6.0 | 14 KB     | ##################################### | 100% \n",
      "perl-bio-featureio-1 | 40 KB     | ##################################### | 100% \n",
      "perl-io-sessiondata- | 8 KB      | ##################################### | 100% \n",
      "perl-cache-cache-1.0 | 31 KB     | ##################################### | 100% \n",
      "perl-pod-usage-1.69  | 13 KB     | ##################################### | 100% \n",
      "perl-app-cpanminus-1 | 234 KB    | ##################################### | 100% \n",
      "perl-class-inspector | 14 KB     | ##################################### | 100% \n",
      "perl-devel-stacktrac | 16 KB     | ##################################### | 100% \n",
      "perl-html-entities-n | 11 KB     | ##################################### | 100% \n",
      "perl-svg-2.84        | 37 KB     | ##################################### | 100% \n",
      "perl-xml-filter-buff | 7 KB      | ##################################### | 100% \n",
      "giflib-5.2.1         | 77 KB     | ##################################### | 100% \n",
      "perl-parent-0.236    | 5 KB      | ##################################### | 100% \n",
      "perl-text-diff-1.45  | 20 KB     | ##################################### | 100% \n",
      "perl-json-xs-2.34    | 63 KB     | ##################################### | 100% \n",
      "vcflib-1.0.0_rc3     | 26.6 MB   | ##################################### | 100% \n",
      "perl-www-robotrules- | 12 KB     | ##################################### | 100% \n",
      "libtool-2.4.6        | 511 KB    | ##################################### | 100% \n",
      "perl-soap-lite-1.19  | 117 KB    | ##################################### | 100% \n",
      "perl-class-load-xs-0 | 13 KB     | ##################################### | 100% \n",
      "perl-data-dumper-2.1 | 35 KB     | ##################################### | 100% \n",
      "libdb-6.1.26         | 21.5 MB   | ##################################### | 100% \n",
      "perl-array-compare-3 | 12 KB     | ##################################### | 100% \n",
      "perl-html-tableextra | 23 KB     | ##################################### | 100% \n",
      "fribidi-1.0.10       | 112 KB    | ##################################### | 100% \n",
      "perl-math-random-0.7 | 55 KB     | ##################################### | 100% \n",
      "perl-xml-xpath-1.44  | 42 KB     | ##################################### | 100% \n",
      "perl-file-which-1.23 | 12 KB     | ##################################### | 100% \n",
      "perl-spreadsheet-par | 63 KB     | ##################################### | 100% \n",
      "perl-business-isbn-3 | 15 KB     | ##################################### | 100% \n",
      "perl-encode-2.88     | 2.1 MB    | ##################################### | 100% \n",
      "perl-mailtools-2.21  | 48 KB     | ##################################### | 100% \n",
      "perl-data-stag-0.14  | 78 KB     | ##################################### | 100% \n",
      "perl-package-stash-x | 21 KB     | ##################################### | 100% \n",
      "perl-libxml-perl-0.0 | 38 KB     | ##################################### | 100% \n",
      "perl-parse-recdescen | 71 KB     | ##################################### | 100% \n",
      "perl-xml-sax-base-1. | 19 KB     | ##################################### | 100% \n",
      "perl-ipc-sharelite-0 | 24 KB     | ##################################### | 100% \n",
      "perl-capture-tiny-0. | 17 KB     | ##################################### | 100% \n",
      "perl-tie-ixhash-1.23 | 11 KB     | ##################################### | 100% \n",
      "perl-libwww-perl-6.3 | 94 KB     | ##################################### | 100% \n",
      "perl-sub-name-0.21   | 13 KB     | ##################################### | 100% \n",
      "perl-encode-locale-1 | 11 KB     | ##################################### | 100% \n",
      "perl-dbd-sqlite-1.64 | 2.3 MB    | ##################################### | 100% \n",
      "perl-data-optlist-0. | 10 KB     | ##################################### | 100% \n",
      "perl-postscript-0.06 | 22 KB     | ##################################### | 100% \n",
      "gdk-pixbuf-2.42.6    | 609 KB    | ##################################### | 100% \n",
      "perl-bio-phylo-0.58  | 310 KB    | ##################################### | 100% \n",
      "perl-aceperl-1.92    | 149 KB    | ##################################### | 100% \n",
      "perl-scalar-list-uti | 38 KB     | ##################################### | 100% \n",
      "samtools-1.14        | 411 KB    | ##################################### | 100% \n",
      "perl-xml-regexp-0.04 | 8 KB      | ##################################### | 100% \n",
      "openjdk-8.0.312      | 97.6 MB   | ##################################### | 100% \n",
      "perl-test-warn-0.36  | 13 KB     | ##################################### | 100% \n",
      "perl-spreadsheet-wri | 259 KB    | ##################################### | 100% \n",
      "perl-package-depreca | 10 KB     | ##################################### | 100% \n",
      "perl-lwp-mediatypes- | 22 KB     | ##################################### | 100% \n",
      "perl-test-1.26       | 14 KB     | ##################################### | 100% \n",
      "perl-getopt-long-2.5 | 27 KB     | ##################################### | 100% \n",
      "tidyp-1.04           | 300 KB    | ##################################### | 100% \n",
      "perl-date-format-2.3 | 29 KB     | ##################################### | 100% \n",
      "perl-digest-sha1-2.1 | 20 KB     | ##################################### | 100% \n",
      "bedtools-2.30.0      | 17.9 MB   | ##################################### | 100% \n",
      "perl-bio-asn1-entrez | 23 KB     | ##################################### | 100% \n",
      "perl-html-parser-3.7 | 66 KB     | ##################################### | 100% \n",
      "perl-test-harness-3. | 82 KB     | ##################################### | 100% \n",
      "perl-convert-binary- | 252 KB    | ##################################### | 100% \n",
      "perl-file-listing-6. | 9 KB      | ##################################### | 100% \n",
      "perl-http-message-6. | 51 KB     | ##################################### | 100% \n",
      "perl-module-runtime- | 7 KB      | ##################################### | 100% \n",
      "perl-unicode-map-0.1 | 254 KB    | ##################################### | 100% \n",
      "perl-xml-writer-0.62 | 18 KB     | ##################################### | 100% \n",
      "k8-0.2.5             | 1.7 MB    | ##################################### | 100% \n",
      "snp-sites-2.5.1      | 41 KB     | ##################################### | 100% \n",
      "perl-svg-graph-0.02  | 26 KB     | ##################################### | 100% \n",
      "perl-xml-twig-3.52   | 144 KB    | ##################################### | 100% \n",
      "perl-mro-compat-0.13 | 10 KB     | ##################################### | 100% \n",
      "samclip-0.4.0        | 7 KB      | ##################################### | 100% \n",
      "perl-mime-tools-5.50 | 113 KB    | ##################################### | 100% \n",
      "perl-image-info-1.38 | 41 KB     | ##################################### | 100% \n",
      "perl-sub-exporter-pr | 8 KB      | ##################################### | 100% \n",
      "perl-eval-closure-0. | 11 KB     | ##################################### | 100% \n",
      "perl-xsloader-0.24   | 8 KB      | ##################################### | 100% \n",
      "libxslt-1.1.33       | 522 KB    | ##################################### | 100% \n",
      "perl-html-element-ex | 24 KB     | ##################################### | 100% \n",
      "perl-bioperl-core-1. | 3.2 MB    | ##################################### | 100% \n",
      "perl-compress-raw-zl | 72 KB     | ##################################### | 100% \n",
      "perl-xml-sax-writer- | 16 KB     | ##################################### | 100% \n",
      "librsvg-2.52.4       | 5.3 MB    | ##################################### | 100% \n",
      "perl-sub-install-0.9 | 10 KB     | ##################################### | 100% \n",
      "perl-template-toolki | 319 KB    | ##################################### | 100% \n",
      "perl-pod-escapes-1.0 | 10 KB     | ##################################### | 100% \n",
      "perl-sub-identify-0. | 12 KB     | ##################################### | 100% \n",
      "perl-digest-hmac-1.0 | 9 KB      | ##################################### | 100% \n",
      "perl-io-string-1.08  | 10 KB     | ##################################### | 100% \n",
      "perl-role-tiny-2.000 | 15 KB     | ##################################### | 100% \n",
      "minimap2-2.23        | 489 KB    | ##################################### | 100% \n",
      "perl-html-formatter- | 30 KB     | ##################################### | 100% \n",
      "perl-math-cdf-0.1    | 69 KB     | ##################################### | 100% \n",
      "perl-timedate-2.30   | 29 KB     | ##################################### | 100% \n",
      "perl-xml-libxml-2.01 | 260 KB    | ##################################### | 100% \n",
      "perl-lwp-simple-6.15 | 348 KB    | ##################################### | 100% \n",
      "perl-xml-parser-2.44 | 151 KB    | ##################################### | 100% \n",
      "perl-lib-0.63        | 6 KB      | ##################################### | 100% \n",
      "perl-math-spline-0.0 | 8 KB      | ##################################### | 100% \n",
      "perl-sort-naturally- | 14 KB     | ##################################### | 100% \n",
      "perl-autoloader-5.74 | 5 KB      | ##################################### | 100% \n",
      "perl-html-tree-5.07  | 129 KB    | ##################################### | 100% \n",
      "snpeff-5.0           | 23.2 MB   | ##################################### | 100% \n",
      "perl-sub-quote-2.006 | 18 KB     | ##################################### | 100% \n",
      "perl-extutils-makema | 153 KB    | ##################################### | 100% \n",
      "perl-carp-1.38       | 15 KB     | ##################################### | 100% \n",
      "perl-bioperl-run-1.0 | 431 KB    | ##################################### | 100% \n",
      "perl-task-weaken-1.0 | 8 KB      | ##################################### | 100% \n",
      "font-ttf-dejavu-sans | 388 KB    | ##################################### | 100% \n",
      "t_coffee-11.0.8      | 33.2 MB   | ##################################### | 100% \n",
      "perl-moo-2.003004    | 38 KB     | ##################################### | 100% \n",
      "perl-image-size-3.30 | 26 KB     | ##################################### | 100% \n",
      "perl-tree-dag_node-1 | 40 KB     | ##################################### | 100% \n",
      "perl-crypt-rc4-2.02  | 8 KB      | ##################################### | 100% \n",
      "perl-pdf-api2-2.035  | 2.3 MB    | ##################################### | 100% \n",
      "pango-1.48.10        | 403 KB    | ##################################### | 100% \n",
      "paml-4.9             | 1.2 MB    | ##################################### | 100% \n",
      "perl-jcode-2.07      | 24 KB     | ##################################### | 100% \n",
      "perl-email-date-form | 8 KB      | ##################################### | 100% \n",
      "perl-file-path-2.16  | 17 KB     | ##################################### | 100% \n",
      "perl-xml-dom-1.46    | 56 KB     | ##################################### | 100% \n",
      "perl-xml-xpathengine | 26 KB     | ##################################### | 100% \n",
      "perl-clone-0.42      | 13 KB     | ##################################### | 100% \n",
      "perl-http-date-6.02  | 10 KB     | ##################################### | 100% \n",
      "font-ttf-ubuntu-0.83 | 1.9 MB    | ##################################### | 100% \n",
      "perl-socket-2.027    | 31 KB     | ##################################### | 100% \n",
      "perl-dbi-1.642       | 549 KB    | ##################################### | 100% \n",
      "gts-0.7.6            | 411 KB    | ##################################### | 100% \n",
      "perl-file-sort-1.01  | 18 KB     | ##################################### | 100% \n",
      "perl-error-0.17027   | 20 KB     | ##################################### | 100% \n",
      "perl-class-data-inhe | 7 KB      | ##################################### | 100% \n",
      "perl-html-tagset-3.2 | 12 KB     | ##################################### | 100% \n",
      "snippy-4.5.0         | 18.9 MB   | ##################################### | 100% \n",
      "perl-apache-test-1.4 | 115 KB    | ##################################### | 100% \n",
      "perl-html-tidy-1.60  | 20 KB     | ##################################### | 100% \n",
      "perl-algorithm-munkr | 11 KB     | ##################################### | 100% \n",
      "perl-type-tiny-1.004 | 119 KB    | ##################################### | 100% \n",
      "perl-mime-base64-3.1 | 15 KB     | ##################################### | 100% \n",
      "perl-appconfig-1.71  | 42 KB     | ##################################### | 100% \n",
      "perl-test-requiresin | 7 KB      | ##################################### | 100% \n",
      "perl-devel-overloadi | 8 KB      | ##################################### | 100% \n",
      "perl-bioperl-1.7.2   | 13 KB     | ##################################### | 100% \n",
      "perl-io-stringy-2.11 | 35 KB     | ##################################### | 100% \n",
      "perl-pathtools-3.75  | 41 KB     | ##################################### | 100% \n",
      "perl-dist-checkconfl | 10 KB     | ##################################### | 100% \n",
      "perl-io-tty-1.12     | 30 KB     | ##################################### | 100% \n",
      "perl-uri-1.76        | 55 KB     | ##################################### | 100% \n",
      "font-ttf-inconsolata | 94 KB     | ##################################### | 100% \n",
      "perl-xml-dom-xpath-0 | 9 KB      | ##################################### | 100% \n",
      "perl-params-util-1.0 | 14 KB     | ##################################### | 100% \n",
      "perl-moose-2.2011    | 444 KB    | ##################################### | 100% \n",
      "perl-ntlm-1.09       | 15 KB     | ##################################### | 100% \n",
      "perl-math-derivative | 9 KB      | ##################################### | 100% \n",
      "perl-xml-sax-expat-0 | 10 KB     | ##################################### | 100% \n",
      "perl-gd-2.68         | 112 KB    | ##################################### | 100% \n",
      "perl-graph-0.9704    | 73 KB     | ##################################### | 100% \n",
      "perl-net-ssleay-1.88 | 289 KB    | ##################################### | 100% \n",
      "perl-types-serialise | 10 KB     | ##################################### | 100% \n",
      "perl-try-tiny-0.30   | 17 KB     | ##################################### | 100% \n",
      "perl-mime-types-2.17 | 56 KB     | ##################################### | 100% \n",
      "perl-bio-tools-run-a | 25 KB     | ##################################### | 100% \n",
      "perl-module-implemen | 9 KB      | ##################################### | 100% \n",
      "perl-dynaloader-1.25 | 3 KB      | ##################################### | 100% \n",
      "perl-file-slurp-tiny | 8 KB      | ##################################### | 100% \n",
      "atk-1.0-2.36.0       | 560 KB    | ##################################### | 100% \n",
      "gsl-2.6              | 3.2 MB    | ##################################### | 100% \n",
      "perl-http-negotiate- | 13 KB     | ##################################### | 100% \n",
      "perl-common-sense-3. | 11 KB     | ##################################### | 100% \n",
      "perl-list-moreutils- | 43 KB     | ##################################### | 100% \n",
      "perl-base-2.23       | 10 KB     | ##################################### | 100% \n",
      "font-ttf-source-code | 684 KB    | ##################################### | 100% \n",
      "perl-digest-perl-md5 | 11 KB     | ##################################### | 100% \n",
      "perl-exporter-tiny-1 | 23 KB     | ##################################### | 100% \n",
      "perl-sub-exporter-0. | 30 KB     | ##################################### | 100% \n",
      "fonts-conda-ecosyste | 4 KB      | ##################################### | 100% \n",
      "any2fasta-0.4.2      | 8 KB      | ##################################### | 100% \n",
      "perl-mldbm-2.05      | 15 KB     | ##################################### | 100% \n",
      "perl-test-leaktrace- | 23 KB     | ##################################### | 100% \n",
      "perl-constant-1.33   | 10 KB     | ##################################### | 100% \n",
      "perl-cgi-4.44        | 116 KB    | ##################################### | 100% \n",
      "perl-convert-binhex- | 25 KB     | ##################################### | 100% \n",
      "perl-graphviz-2.24   | 36 KB     | ##################################### | 100% \n",
      "clustalw-2.1         | 318 KB    | ##################################### | 100% \n",
      "bcftools-1.14        | 911 KB    | ##################################### | 100% \n",
      "perl-set-scalar-1.29 | 19 KB     | ##################################### | 100% \n",
      "perl-devel-globaldes | 7 KB      | ##################################### | 100% \n",
      "libwebp-1.2.1        | 84 KB     | ##################################### | 100% \n",
      "perl-xml-simple-2.25 | 48 KB     | ##################################### | 100% \n",
      "perl-bio-coordinate- | 35 KB     | ##################################### | 100% \n",
      "perl-test-difference | 15 KB     | ##################################### | 100% \n",
      "perl-json-4.02       | 52 KB     | ##################################### | 100% \n",
      "perl-algorithm-diff- | 34 KB     | ##################################### | 100% \n",
      "libdeflate-1.7       | 67 KB     | ##################################### | 100% \n",
      "curl-7.80.0          | 153 KB    | ##################################### | 100% \n",
      "perl-exception-class | 19 KB     | ##################################### | 100% \n",
      "perl-db-file-1.852   | 52 KB     | ##################################### | 100% \n",
      "perl-exporter-5.72   | 13 KB     | ##################################### | 100% \n",
      "perl-xml-namespacesu | 11 KB     | ##################################### | 100% \n",
      "perl-statistics-desc | 32 KB     | ##################################### | 100% \n",
      "perl-business-isbn-d | 15 KB     | ##################################### | 100% \n",
      "vt-2015.11.10        | 1.5 MB    | ##################################### | 100% \n",
      "perl-package-stash-0 | 66 KB     | ##################################### | 100% \n",
      "perl-font-afm-1.20   | 14 KB     | ##################################### | 100% \n",
      "freebayes-1.3.2      | 16.3 MB   | ##################################### | 100% \n",
      "perl-bio-samtools-1. | 172 KB    | ##################################### | 100% \n",
      "perl-mime-lite-3.030 | 51 KB     | ##################################### | 100% \n",
      "perl-test-exception- | 12 KB     | ##################################### | 100% \n",
      "gtk2-2.24.33         | 7.3 MB    | ##################################### | 100% \n",
      "perl-xml-libxslt-1.9 | 40 KB     | ##################################### | 100% \n",
      "perl-xml-sax-1.02    | 36 KB     | ##################################### | 100% \n",
      "perl-module-runtime- | 15 KB     | ##################################### | 100% \n",
      "perl-list-moreutils- | 28 KB     | ##################################### | 100% \n",
      "perl-lwp-protocol-ht | 9 KB      | ##################################### | 100% \n",
      "perl-font-ttf-1.06   | 200 KB    | ##################################### | 100% \n",
      "perl-net-http-6.19   | 19 KB     | ##################################### | 100% \n",
      "bwa-0.7.17           | 608 KB    | ##################################### | 100% \n",
      "perl-test-most-0.35  | 18 KB     | ##################################### | 100% \n",
      "perl-bio-tools-run-a | 24 KB     | ##################################### | 100% \n",
      "perl-file-temp-0.230 | 26 KB     | ##################################### | 100% \n",
      "perl-http-cookies-6. | 19 KB     | ##################################### | 100% \n",
      "libtiff-4.3.0        | 668 KB    | ##################################### | 100% \n",
      "perl-ole-storage_lit | 18 KB     | ##################################### | 100% \n",
      "perl-bio-tools-phylo | 57 KB     | ##################################### | 100% \n",
      "perl-digest-md5-2.55 | 18 KB     | ##################################### | 100% \n",
      "perl-storable-3.15   | 63 KB     | ##################################### | 100% \n",
      "perl-time-local-1.28 | 10 KB     | ##################################### | 100% \n",
      "perl-time-hires-1.97 | 25 KB     | ##################################### | 100% \n",
      "perl-sub-uplevel-0.2 | 13 KB     | ##################################### | 100% \n",
      "perl-html-treebuilde | 12 KB     | ##################################### | 100% \n",
      "parallel-20211022    | 1.8 MB    | ##################################### | 100% \n",
      "perl-io-socket-ssl-2 | 151 KB    | ##################################### | 100% \n",
      "perl-test-deep-1.128 | 36 KB     | ##################################### | 100% \n",
      "perl-class-load-0.25 | 12 KB     | ##################################### | 100% \n",
      "perl-ipc-run-2018052 | 74 KB     | ##################################### | 100% \n",
      "lerc-2.2.1           | 213 KB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: / \n",
      "- \n",
      "done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install snippy==4.05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7af4fae4-8cc6-4824-9a83-8720de58d569",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - swarm==3.0.0=hc9558a2_0\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    arpack-3.7.0               |       hdefa2d7_2         215 KB  conda-forge\n",
      "    glpk-4.65                  |    h9202a9a_1004         1.0 MB  conda-forge\n",
      "    metis-5.1.0                |    h58526e2_1006         4.1 MB  conda-forge\n",
      "    mpfr-4.1.0                 |       h9202a9a_1         2.6 MB  conda-forge\n",
      "    python-igraph-0.9.8        |   py37h6c76e3a_4         1.3 MB  conda-forge\n",
      "    suitesparse-5.10.1         |       h9e50725_1         2.4 MB  conda-forge\n",
      "    swarm-3.0.0                |       hc9558a2_0         194 KB  bioconda\n",
      "    tbb-2021.4.0               |       h4bd325d_1         2.0 MB  conda-forge\n",
      "    texttable-1.6.4            |     pyhd8ed1ab_0          16 KB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        13.8 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  arpack             conda-forge/linux-64::arpack-3.7.0-hdefa2d7_2\n",
      "  glpk               conda-forge/linux-64::glpk-4.65-h9202a9a_1004\n",
      "  metis              conda-forge/linux-64::metis-5.1.0-h58526e2_1006\n",
      "  mpfr               conda-forge/linux-64::mpfr-4.1.0-h9202a9a_1\n",
      "  python-igraph      conda-forge/linux-64::python-igraph-0.9.8-py37h6c76e3a_4\n",
      "  suitesparse        conda-forge/linux-64::suitesparse-5.10.1-h9e50725_1\n",
      "  swarm              bioconda/linux-64::swarm-3.0.0-hc9558a2_0\n",
      "  tbb                conda-forge/linux-64::tbb-2021.4.0-h4bd325d_1\n",
      "  texttable          conda-forge/noarch::texttable-1.6.4-pyhd8ed1ab_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "arpack-3.7.0         | 215 KB    | ##################################### | 100% \n",
      "swarm-3.0.0          | 194 KB    | ##################################### | 100% \n",
      "texttable-1.6.4      | 16 KB     | ##################################### | 100% \n",
      "metis-5.1.0          | 4.1 MB    | ##################################### | 100% \n",
      "mpfr-4.1.0           | 2.6 MB    | ##################################### | 100% \n",
      "suitesparse-5.10.1   | 2.4 MB    | ##################################### | 100% \n",
      "python-igraph-0.9.8  | 1.3 MB    | ##################################### | 100% \n",
      "glpk-4.65            | 1.0 MB    | ##################################### | 100% \n",
      "tbb-2021.4.0         | 2.0 MB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c bioconda swarm=3.0.0=hc9558a2_0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37d780ed-c13a-4517-87dc-98890ca7dae3",
   "metadata": {},
   "source": [
    "Check the instalation by asking for swarm help:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0f345976-2fae-4a2f-a51d-0b285dd21cbd",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Swarm 3.0.0\n",
      "Copyright (C) 2012-2019 Torbjorn Rognes and Frederic Mahe\n",
      "https://github.com/torognes/swarm\n",
      "\n",
      "Mahe F, Rognes T, Quince C, de Vargas C, Dunthorn M (2014)\n",
      "Swarm: robust and fast clustering method for amplicon-based studies\n",
      "PeerJ 2:e593 https://doi.org/10.7717/peerj.593\n",
      "\n",
      "Mahe F, Rognes T, Quince C, de Vargas C, Dunthorn M (2015)\n",
      "Swarm v2: highly-scalable and high-resolution amplicon clustering\n",
      "PeerJ 3:e1420 https://doi.org/10.7717/peerj.1420\n",
      "\n",
      "Usage: swarm [OPTIONS] [FASTAFILE]\n",
      "\n",
      "General options:\n",
      " -h, --help                          display this help and exit\n",
      " -t, --threads INTEGER               number of threads to use (1)\n",
      " -v, --version                       display version information and exit\n",
      "\n",
      "Clustering options:\n",
      " -d, --differences INTEGER           resolution (1)\n",
      " -n, --no-otu-breaking               never break OTUs (not recommended!)\n",
      "\n",
      "Fastidious options (only when d = 1):\n",
      " -b, --boundary INTEGER              min mass of large OTUs (3)\n",
      " -c, --ceiling INTEGER               max memory in MB for Bloom filter (unlim.)\n",
      " -f, --fastidious                    link nearby low-abundance swarms\n",
      " -y, --bloom-bits INTEGER            bits used per Bloom filter entry (16)\n",
      "\n",
      "Input/output options:\n",
      " -a, --append-abundance INTEGER      value to use when abundance is missing\n",
      " -i, --internal-structure FILENAME   write internal OTU structure to file\n",
      " -l, --log FILENAME                  log to file, not to stderr\n",
      " -o, --output-file FILENAME          output result to file (stdout)\n",
      " -r, --mothur                        output using mothur-like format\n",
      " -s, --statistics-file FILENAME      dump OTU statistics to file\n",
      " -u, --uclust-file FILENAME          output using UCLUST-like format to file\n",
      " -w, --seeds FILENAME                write OTU representatives to FASTA file\n",
      " -z, --usearch-abundance             abundance annotation in usearch style\n",
      "\n",
      "Pairwise alignment advanced options (only when d > 1):\n",
      " -m, --match-reward INTEGER          reward for nucleotide match (5)\n",
      " -p, --mismatch-penalty INTEGER      penalty for nucleotide mismatch (4)\n",
      " -g, --gap-opening-penalty INTEGER   gap open penalty (12)\n",
      " -e, --gap-extension-penalty INTEGER gap extension penalty (4)\n",
      "\n",
      "See 'man swarm' for more details.\n"
     ]
    }
   ],
   "source": [
    "!swarm -h | head "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f55d8cc-bb1b-4edd-bce1-ea5f16f8ceeb",
   "metadata": {},
   "source": [
    "<p>Install <strong>Pear</strong>, which is the tool adoped for merging the Illumina paired-end reads</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71457a7c-d1d2-4e50-b7e7-0c53bcfb7305",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - pear==0.9.6=h2f06484_6\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    pear-0.9.6                 |       h2f06484_6          65 KB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:          65 KB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  pear               bioconda/linux-64::pear-0.9.6-h2f06484_6\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "pear-0.9.6           | 65 KB     | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c bioconda pear=0.9.6=h2f06484_6"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b68d7e6-1222-4484-b7f8-626a3878b4cc",
   "metadata": {},
   "source": [
    "Ask for Pear -h to check the instalation:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32902a7e-8dc0-47a7-9dea-edeb6b172346",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "PEAR v0.9.6 [January 15, 2015]  - [+bzlib +zlib]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "License: Creative Commons Licence\n",
      "Bug-reports and requests to: Tomas.Flouri@h-its.org and Jiajie.Zhang@h-its.org\n",
      "\n",
      "\n",
      "Usage: pear <options>\n",
      "Standard (mandatory):\n",
      "  -f, --forward-fastq         <str>     Forward paired-end FASTQ file.\n",
      "  -r, --reverse-fastq         <str>     Reverse paired-end FASTQ file.\n",
      "  -o, --output                <str>     Output filename.\n",
      "Optional:\n",
      "  -p, --p-value               <float>   Specify  a p-value for the statistical test. If the computed\n",
      "                                        p-value of a possible assembly exceeds the specified p-value\n",
      "                                        then  paired-end  read  will not be assembled. Valid options\n",
      "                                        are: 0.0001, 0.001, 0.01, 0.05 and 1.0. Setting 1.0 disables\n",
      "                                        the test. (default: 0.01)\n",
      "  -v, --min-overlap           <int>     Specify the minimum overlap size. The minimum overlap may be\n",
      "                                        set to 1 when the statistical test is used. However, further\n",
      "                                        restricting  the  minimum overlap size to a proper value may\n",
      "                                        reduce false-positive assembles. (default: 10)\n",
      "  -m, --max-assembly-length   <int>     Specify   the  maximum  possible  length  of  the  assembled\n",
      "                                        sequences.  Setting this value to 0 disables the restriction\n",
      "                                        and assembled sequences may be arbitrary long. (default: 0)\n",
      "  -n, --min-assembly-length   <int>     Specify   the  minimum  possible  length  of  the  assembled\n",
      "                                        sequences.  Setting this value to 0 disables the restriction\n",
      "                                        and  assembled  sequences  may be arbitrary short. (default:\n",
      "                                        50)\n",
      "  -t, --min-trim-length       <int>     Specify  the  minimum length of reads after trimming the low\n",
      "                                        quality part (see option -q). (default: 1)\n",
      "  -q, --quality-threshold     <int>     Specify  the  quality  score  threshold for trimming the low\n",
      "                                        quality  part  of  a  read.  If  the  quality  scores of two\n",
      "                                        consecutive  bases  are  strictly  less  than  the specified\n",
      "                                        threshold,  the  rest of the read will be trimmed. (default:\n",
      "                                        0)\n",
      "  -u, --max-uncalled-base     <float>   Specify  the maximal proportion of uncalled bases in a read.\n",
      "                                        Setting this value to 0 will cause PEAR to discard all reads\n",
      "                                        containing  uncalled  bases.  The other extreme setting is 1\n",
      "                                        which  causes  PEAR  to process all reads independent on the\n",
      "                                        number of uncalled bases. (default: 1)\n",
      "  -g, --test-method           <int>     Specify  the  type  of  statistical  test.  Two  options are\n",
      "                                        available. (default: 1)\n",
      "                                        1: Given the minimum allowed overlap, test using the highest\n",
      "                                        OES. Note that due to its discrete nature, this test usually\n",
      "                                        yields  a lower p-value for the assembled read than the cut-\n",
      "                                        off  (specified  by -p). For example, setting the cut-off to\n",
      "                                        0.05  using  this  test,  the  assembled reads might have an\n",
      "                                        actual p-value of 0.02.\n",
      "\n",
      "                                        2. Use the acceptance probability (m.a.p). This test methods\n",
      "                                        computes  the same probability as test method 1. However, it\n",
      "                                        assumes  that  the  minimal  overlap is the observed overlap\n",
      "                                        with  the  highest  OES, instead of the one specified by -v.\n",
      "                                        Therefore,  this  is  not  a  valid statistical test and the\n",
      "                                        'p-value'  is  in fact the maximal probability for accepting\n",
      "                                        the assembly. Nevertheless, we observed in practice that for\n",
      "                                        the case the actual overlap sizes are relatively small, test\n",
      "                                        2  can  correctly  assemble  more  reads  with only slightly\n",
      "                                        higher false-positive rate.\n",
      "  -e, --empirical-freqs                 Disable  empirical base frequencies. (default: use empirical\n",
      "                                        base frequencies)\n",
      "  -s, --score-method          <int>     Specify the scoring method. (default: 2)\n",
      "                                        1. OES with +1 for match and -1 for mismatch.\n",
      "                                        2: Assembly score (AS). Use +1 for match and -1 for mismatch\n",
      "                                        multiplied by base quality scores.\n",
      "                                        3: Ignore quality scores and use +1 for a match and -1 for a\n",
      "                                        mismatch.\n",
      "  -b, --phred-base            <int>     Base PHRED quality score. (default: 33)\n",
      "  -y, --memory                <str>     Specify  the  amount of memory to be used. The number may be\n",
      "                                        followed  by  one  of  the  letters  K,  M,  or  G  denoting\n",
      "                                        Kilobytes,  Megabytes and Gigabytes, respectively. Bytes are\n",
      "                                        assumed in case no letter is specified.\n",
      "  -c, --cap                   <int>     Specify  the upper bound for the resulting quality score. If\n",
      "                                        set to zero, capping is disabled. (default: 40)\n",
      "  -j, --threads               <int>     Number of threads to use\n",
      "  -z, --nbase                           When  merging  a  base-pair  that  consists of two non-equal\n",
      "                                        bases  out  of which none is degenerate, set the merged base\n",
      "                                        to N and use the highest quality score of the two bases\n",
      "  -h, --help                            This help screen.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!pear -h"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7c75e91-5c92-458b-919b-2fc2e456d325",
   "metadata": {},
   "source": [
    "<p>Install <strong>Taxonkit</strong>, which is the tool adoped for getting the lineage for the references databases, and download the taxdump and nucleotide to accession files from NCBI.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04b172f7-9983-4f37-be5f-5b559e01be35",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - taxonkit\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    taxonkit-0.9.0             |       h9ee0642_0         2.5 MB  bioconda\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         2.5 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  taxonkit           bioconda/linux-64::taxonkit-0.9.0-h9ee0642_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "taxonkit-0.9.0       | 2.5 MB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c bioconda taxonkit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94fb08ff-87f6-4703-8e6a-b4870f90bdf8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-12-09 18:10:14--  https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/nucl_gb.accession2taxid.gz\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 130.14.250.11, 130.14.250.13, 2607:f220:41e:250::10, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|130.14.250.11|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 2133379214 (2.0G) [application/x-gzip]\n",
      "Saving to: nucl_gb.accession2taxid.gz\n",
      "\n",
      "nucl_gb.accession2t 100%[===================>]   1.99G  50.1MB/s    in 68s     \n",
      "\n",
      "2021-12-09 18:11:23 (29.8 MB/s) - nucl_gb.accession2taxid.gz saved [2133379214/2133379214]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/nucl_gb.accession2taxid.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e17e7ab9-2b2c-4a1f-93fd-52c5b80f0b93",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-12-09 18:11:23--  ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz\n",
      "           => taxdump.tar.gz\n",
      "Resolving ftp.ncbi.nih.gov (ftp.ncbi.nih.gov)... 165.112.9.228, 165.112.9.229, 2607:f220:41e:250::10, ...\n",
      "Connecting to ftp.ncbi.nih.gov (ftp.ncbi.nih.gov)|165.112.9.228|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /pub/taxonomy ... done.\n",
      "==> SIZE taxdump.tar.gz ... 56823744\n",
      "==> PASV ... done.    ==> RETR taxdump.tar.gz ... done.\n",
      "Length: 56823744 (54M) (unauthoritative)\n",
      "\n",
      "taxdump.tar.gz      100%[===================>]  54.19M  32.8MB/s    in 1.7s    \n",
      "\n",
      "2021-12-09 18:11:25 (32.8 MB/s) - taxdump.tar.gz saved [56823744]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2fb38441-a80d-40ca-9840-dca946b322c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "citations.dmp\n",
      "delnodes.dmp\n",
      "division.dmp\n",
      "gencode.dmp\n",
      "merged.dmp\n",
      "names.dmp\n",
      "nodes.dmp\n",
      "gc.prt\n",
      "readme.txt\n"
     ]
    }
   ],
   "source": [
    "!tar -zxvf taxdump.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb87c1ca-ea45-4900-9ee8-8795b75b0bb1",
   "metadata": {},
   "outputs": [],
   "source": [
    "%mkdir -p $HOME/.taxonkit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "78480b86-823e-4626-a648-ccf45206c6c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "%cp names.dmp nodes.dmp delnodes.dmp merged.dmp $HOME/.taxonkit"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cfabe374-64b6-41aa-9dad-15c6a03669dc",
   "metadata": {},
   "source": [
    "Check the installation asking for taxonkit -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ab73e8d-db50-4fd7-a582-d1f8e0dba278",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TaxonKit - A Practical and Efficient NCBI Taxonomy Toolkit\n",
      "\n",
      "Version: 0.9.0\n",
      "\n",
      "Author: Wei Shen <shenwei356@gmail.com>\n",
      "\n",
      "Source code: https://github.com/shenwei356/taxonkit\n",
      "Documents  : https://bioinf.shenwei.me/taxonkit\n",
      "Citation   : https://www.sciencedirect.com/science/article/pii/S1673852721000837\n",
      "\n",
      "Dataset:\n",
      "\n",
      "    Please download and uncompress \"taxdump.tar.gz\":\n",
      "    ftp://ftp.ncbi.nih.gov/pub/taxonomy/taxdump.tar.gz\n",
      "\n",
      "    and copy \"names.dmp\", \"nodes.dmp\", \"delnodes.dmp\" and \"merged.dmp\" to data directory:\n",
      "    \"/home/jupyter/.taxonkit\"\n",
      "\n",
      "    or some other directory, and later you can refer to using flag --data-dir,\n",
      "    or environment variable TAXONKIT_DB\n",
      "\n",
      "Usage:\n",
      "  taxonkit [command]\n",
      "\n",
      "Available Commands:\n",
      "  filter          Filter TaxIds by taxonomic rank range\n",
      "  genautocomplete generate shell autocompletion script (bash|zsh|fish|powershell)\n",
      "  lca             Compute lowest common ancestor (LCA) for TaxIds\n",
      "  lineage         Query taxonomic lineage of given TaxIds\n",
      "  list            List taxonomic subtrees of given TaxIds\n",
      "  name2taxid      Convert scientific names to TaxIds\n",
      "  profile2cami    Convert metagenomic profile table to CAMI format\n",
      "  reformat        Reformat lineage in canonical ranks\n",
      "  taxid-changelog Create TaxId changelog from dump archives\n",
      "  version         print version information and check for update\n",
      "\n",
      "Flags:\n",
      "      --data-dir string   directory containing nodes.dmp and names.dmp (default \"/home/jupyter/.taxonkit\")\n",
      "  -h, --help              help for taxonkit\n",
      "      --line-buffered     use line buffering on output, i.e., immediately writing to stdin/file for every line of output\n",
      "  -o, --out-file string   out file (\"-\" for stdout, suffix .gz for gzipped out) (default \"-\")\n",
      "  -j, --threads int       number of CPUs. 4 is enough (default 4)\n",
      "      --verbose           print verbose information\n",
      "\n",
      "Use \"taxonkit [command] --help\" for more information about a command.\n"
     ]
    }
   ],
   "source": [
    "!taxonkit -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aca6692c-d00b-4707-a64c-7feb8abdac47",
   "metadata": {},
   "outputs": [],
   "source": [
    "%rm *.dmp taxdump.tar.gz readme.txt gc.prt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a718b7e-5976-428b-bcda-6541eda76cfd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1_merge_pear.sh                         Install_Setup.ipynb\n",
      "2a_preparing_tags_LCPI.pl               NCBI_lineage_dups.pl\n",
      "2b_demulti_dual_index_linked.sh         NCBI_lineage_edit.pl\n",
      "3_tag_primer_clipping_clean_derep.sh    README.md\n",
      "4_clustering_swarm_with_chim_filter.sh  count_fastq_sequences.sh\n",
      "5_BLAST_options.sh                      metapipe-r.ipynb\n",
      "6a_OTU_contingency_table.py             metapipe-run.ipynb\n",
      "6a_build_OTU_table.sh                   nucl_gb.accession2taxid.gz\n",
      "6b_filter_OTU_table.sh                  trnL_S1_L001_R1_001_sub.fastq\n",
      "DB1_NCBI_download_refs.sh               trnL_S1_L001_R2_001_sub.fastq\n",
      "DB2_primers_cut.sh                      trnL_sub_metadata.txt\n",
      "DB3_NCBI_lineage.sh                     trnL_sub_tags.txt\n"
     ]
    }
   ],
   "source": [
    "%ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30f55e71-c063-433e-80dc-c247a20ecce0",
   "metadata": {},
   "source": [
    "<p>Install <strong>BLAST</strong>, which is the tool adoped for assigning taxonomy, and download the taxdb file from NCBI.</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "62ea4a50-c3a3-484b-8a8d-27430a007652",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda install -c bioconda blast"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "74fa4afd-2a76-4dd2-942a-5cee8a3dcfd5",
   "metadata": {},
   "source": [
    "Ckeck the installation by asking blast help:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "638317c2-b13e-4924-9beb-7db1d6ba1023",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "USAGE\n",
      "  blastn [-h] [-help] [-import_search_strategy filename]\n",
      "    [-export_search_strategy filename] [-task task_name] [-db database_name]\n",
      "    [-dbsize num_letters] [-gilist filename] [-seqidlist filename]\n",
      "    [-negative_gilist filename] [-negative_seqidlist filename]\n",
      "    [-taxids taxids] [-negative_taxids taxids] [-taxidlist filename]\n",
      "    [-negative_taxidlist filename] [-entrez_query entrez_query]\n",
      "    [-db_soft_mask filtering_algorithm] [-db_hard_mask filtering_algorithm]\n",
      "    [-subject subject_input_file] [-subject_loc range] [-query input_file]\n",
      "    [-out output_file] [-evalue evalue] [-word_size int_value]\n",
      "    [-gapopen open_penalty] [-gapextend extend_penalty]\n",
      "    [-perc_identity float_value] [-qcov_hsp_perc float_value]\n",
      "    [-max_hsps int_value] [-xdrop_ungap float_value] [-xdrop_gap float_value]\n",
      "    [-xdrop_gap_final float_value] [-searchsp int_value]\n",
      "    [-sum_stats bool_value] [-penalty penalty] [-reward reward] [-no_greedy]\n",
      "    [-min_raw_gapped_score int_value] [-template_type type]\n",
      "    [-template_length int_value] [-dust DUST_options]\n",
      "    [-filtering_db filtering_database]\n",
      "    [-window_masker_taxid window_masker_taxid]\n",
      "    [-window_masker_db window_masker_db] [-soft_masking soft_masking]\n",
      "    [-ungapped] [-culling_limit int_value] [-best_hit_overhang float_value]\n",
      "    [-best_hit_score_edge float_value] [-subject_besthit]\n",
      "    [-window_size int_value] [-off_diagonal_range int_value]\n",
      "    [-use_index boolean] [-index_name string] [-lcase_masking]\n",
      "    [-query_loc range] [-strand strand] [-parse_deflines] [-outfmt format]\n",
      "    [-show_gis] [-num_descriptions int_value] [-num_alignments int_value]\n",
      "    [-line_length line_length] [-html] [-sorthits sort_hits]\n",
      "    [-sorthsps sort_hsps] [-max_target_seqs num_sequences]\n",
      "    [-num_threads int_value] [-mt_mode int_value] [-remote] [-version]\n",
      "\n",
      "DESCRIPTION\n",
      "   Nucleotide-Nucleotide BLAST 2.12.0+\n",
      "\n",
      "Use '-help' to print detailed descriptions of command line arguments\n"
     ]
    }
   ],
   "source": [
    "!blastn -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "363edecf-6d26-4cde-9d43-56f7ea519f17",
   "metadata": {},
   "outputs": [],
   "source": [
    "!wget https://ftp.ncbi.nlm.nih.gov/blast/db/taxdb.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51c9e4e7-1bb6-4301-af86-171083883c69",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "tar -zxvf taxdb.tar.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d224c3f0-7e44-45a0-9044-ecfb1f5d4e1c",
   "metadata": {},
   "source": [
    "<p>Install <strong>entrez-direct&nbsp;</strong>toolkit, which is the tool adoped for searching and downloading NCBI references</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60b562be-3368-4b33-be62-5b0806e22a66",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda install -c bioconda entrez-direct"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5291fa3-3de0-4915-97ad-696f002ca5cb",
   "metadata": {},
   "source": [
    "Check the installation by asking entrez-direct help:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a918687c-cb5f-410f-8460-b3cdc2434914",
   "metadata": {},
   "outputs": [],
   "source": [
    "!esearch -h | head"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ec29c91-9337-4745-aed0-aaf6cf75bba9",
   "metadata": {},
   "source": [
    "<p>Install <strong>mamba&nbsp;</strong>and create a <strong>Python 2.7</strong> environment, which is necessary for running the &apos;build_OTU_table.sh script</p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab3619c6-0cb7-412d-a549-661f1ab5d50b",
   "metadata": {},
   "outputs": [],
   "source": [
    "conda install -y -c conda-forge mamba"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51c5a9be-bfab-4c4e-bc1d-0b3d214f68a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mamba create -y -n py27 python=2.7 ipykernel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "5dd30dc6-9102-4ad5-be97-8174a25918d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Python 2.7.15\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!conda run -n py27 python --version"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42b77371-b1d0-4554-a59b-001e5e5a6e5a",
   "metadata": {},
   "source": [
    "https://console.cloud.google.com/marketplace/product/rstudio-launcher-public/rstudio-server-pro-standard-for-gcp?q=rstudio&id=76ae9647-2d08-4888-b737-74f751581848&project=metapipe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "21e4e0a0-81d3-4451-9047-a7027dc4245c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://metapipe-data/FINAL/trnL_sub_tags.txt...\n",
      "/ [1 files][  1.5 KiB/  1.5 KiB]                                                \n",
      "Operation completed over 1 objects/1.5 KiB.                                      \n"
     ]
    }
   ],
   "source": [
    "!gsutil cp gs://metapipe-data/FINAL/trnL_sub_tags.txt ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "7a2544bb-646a-4752-9155-c5e2b44764f2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://metapipe-files/trnL_sub_metadata.txt...\n",
      "/ [1 files][  2.6 KiB/  2.6 KiB]                                                \n",
      "Operation completed over 1 objects/2.6 KiB.                                      \n"
     ]
    }
   ],
   "source": [
    "!gsutil cp gs://metapipe-files/trnL_sub_metadata.txt ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "17993d00-cd01-4903-b72d-4eb6024c4265",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://metapipe-data/RUN_ON_JUPYTER/trnL_S1_L001_R1_001_sub.fastq...\n",
      "/ [1 files][367.7 KiB/367.7 KiB]                                                \n",
      "Operation completed over 1 objects/367.7 KiB.                                    \n"
     ]
    }
   ],
   "source": [
    "!gsutil cp gs://metapipe-data/RUN_ON_JUPYTER/trnL_S1_L001_R1_001_sub.fastq ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "75e6c6b1-786d-40b0-821f-74bb60ac8a85",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying gs://metapipe-data/RUN_ON_JUPYTER/trnL_S1_L001_R2_001_sub.fastq...\n",
      "/ [1 files][367.7 KiB/367.7 KiB]                                                \n",
      "Operation completed over 1 objects/367.7 KiB.                                    \n"
     ]
    }
   ],
   "source": [
    "!gsutil cp gs://metapipe-data/RUN_ON_JUPYTER/trnL_S1_L001_R2_001_sub.fastq ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9efdd9ff-5cc5-43df-abf9-437543c8109e",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gsutil cp gs://metapipe-files/dioscorea.png ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce729e45-affc-42ba-a0a2-ce70c4cb8825",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gsutil cp gs://metapipe-files/discorea.png ."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "847c3a1d-a064-48e2-aef2-fe4ccf147f98",
   "metadata": {},
   "source": [
    "https://docs.rstudio.com/rsw/integration/jupyter-standalone/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ac5c7d5-c736-40b3-b7c9-1181e13bea0f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!sudo apt-get -y update"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "174a59a6-7949-43c1-b6ba-650d303ce991",
   "metadata": {},
   "outputs": [],
   "source": [
    "!sudo apt-get install -y gdebi-core"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b4377827-c8a3-4b7b-bc69-45c323b3b03c",
   "metadata": {},
   "outputs": [],
   "source": [
    "!curl -O https://cdn.rstudio.com/r/debian-10/pkgs/r-4.0.5_1_amd64.deb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "858ee444-2afe-455f-9efb-6688e525bd58",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!sudo gdebi --n r-4.0.5_1_amd64.deb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "58f4c97c-8cfc-42ad-909a-ee0b89186981",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R version 4.0.5 (2021-03-31) -- \"Shake and Throw\"\n",
      "Copyright (C) 2021 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under the terms of the\n",
      "GNU General Public License versions 2 or 3.\n",
      "For more information about these matters see\n",
      "https://www.gnu.org/licenses/.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/opt/R/4.0.5/bin/R --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b8d9a31-58b7-4235-b418-fabb6bda4093",
   "metadata": {},
   "outputs": [],
   "source": [
    "!sudo ln -s /opt/R/4.0.5/bin/R /usr/local/bin/R\n",
    "!sudo ln -s /opt/R/4.0.5/bin/Rscript /usr/local/bin/Rscript"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4065fd4a-a4bc-465a-a9d7-fc24dae32e96",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "!sudo apt-get install gdebi-core\n",
    "!curl -O https://download2.rstudio.org/server/bionic/amd64/rstudio-workbench-2021.09.0-351.pro6-amd64.deb\n",
    "!sudo gdebi --n rstudio-workbench-2021.09.0-351.pro6-amd64.deb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "a26801ba-ea64-44e8-8737-f11b9ce7c82e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RStudio License Manager 2021.09.0+351.pro6\n",
      "\n",
      "-- Local license status --\n",
      "\n",
      "Trial-Type: Verified\n",
      "Status: Evaluation\n",
      "Days-Left: 45\n",
      "License-Scope: System\n",
      "License-Engine: 4.4.3.0\n",
      "\n",
      "-- Floating license status --\n",
      "\n",
      "License server not in use.\n"
     ]
    }
   ],
   "source": [
    "!sudo rstudio-server license-manager status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39087694-6b8b-47f6-8691-518cc4b4e4eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# After R installation we need to setup its comunication with Jupyter using the R kernel\n",
    "R -e \"install.packages('IRkernel', repos='http://cran.us.r-project.org')\"\n",
    "R -e \"IRkernel::installspec()\"\n",
    "# After the installation it is required to refresh the notebook kernel by simply refreshing the page (F5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "716d8dfe-eb7a-4d67-bcc9-cd11cd358438",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: failed with initial frozen solve. Retrying with flexible solve.\n",
      "Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.\n",
      "Collecting package metadata (repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - bioconductor-phyloseq\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    _r-mutex-1.0.1             |      anacondar_1           3 KB  conda-forge\n",
      "    binutils_impl_linux-64-2.36.1|       h193b22a_2        10.4 MB  conda-forge\n",
      "    binutils_linux-64-2.36     |       hf3e587d_2          23 KB  conda-forge\n",
      "    bioconductor-biobase-2.54.0|    r41hd029910_0         2.3 MB  bioconda\n",
      "    bioconductor-biocgenerics-0.40.0|    r41hdfd78af_0         631 KB  bioconda\n",
      "    bioconductor-biomformat-1.22.0|    r41hdfd78af_0         542 KB  bioconda\n",
      "    bioconductor-biostrings-2.62.0|    r41hd029910_0        13.9 MB  bioconda\n",
      "    bioconductor-genomeinfodb-1.30.0|    r41hdfd78af_0         4.1 MB  bioconda\n",
      "    bioconductor-genomeinfodbdata-1.2.7|    r41hdfd78af_1           7 KB  bioconda\n",
      "    bioconductor-iranges-2.28.0|    r41hd029910_0         2.4 MB  bioconda\n",
      "    bioconductor-multtest-2.50.0|    r41hd029910_0         889 KB  bioconda\n",
      "    bioconductor-phyloseq-1.38.0|    r41hdfd78af_0         5.7 MB  bioconda\n",
      "    bioconductor-rhdf5-2.38.0  |    r41hfe70e90_1         2.8 MB  bioconda\n",
      "    bioconductor-rhdf5filters-1.6.0|    r41h399db7b_0         664 KB  bioconda\n",
      "    bioconductor-rhdf5lib-1.16.0|    r41hd029910_0         3.9 MB  bioconda\n",
      "    bioconductor-s4vectors-0.32.0|    r41hd029910_0         2.2 MB  bioconda\n",
      "    bioconductor-xvector-0.34.0|    r41hd029910_0         726 KB  bioconda\n",
      "    bioconductor-zlibbioc-1.40.0|    r41hd029910_0         114 KB  bioconda\n",
      "    bwidget-1.9.14             |       ha770c72_1         120 KB  conda-forge\n",
      "    gcc_impl_linux-64-9.4.0    |      h03d3576_11        45.6 MB  conda-forge\n",
      "    gcc_linux-64-9.4.0         |       h391b98a_2          24 KB  conda-forge\n",
      "    gfortran_impl_linux-64-9.4.0|      h0003116_11        14.6 MB  conda-forge\n",
      "    gfortran_linux-64-9.4.0    |       hf0ab688_2          24 KB  conda-forge\n",
      "    gxx_impl_linux-64-9.4.0    |      h03d3576_11        10.6 MB  conda-forge\n",
      "    gxx_linux-64-9.4.0         |       h0316aca_2          24 KB  conda-forge\n",
      "    kernel-headers_linux-64-2.6.32|      he073ed8_15         707 KB  conda-forge\n",
      "    libgcc-devel_linux-64-9.4.0|      hd854feb_11         4.0 MB  conda-forge\n",
      "    libsanitizer-9.4.0         |      h79bfe98_11         6.9 MB  conda-forge\n",
      "    libstdcxx-devel_linux-64-9.4.0|      hd854feb_11         9.9 MB  conda-forge\n",
      "    make-4.3                   |       hd18ef5c_1         507 KB  conda-forge\n",
      "    pcre2-10.37                |       h032f7d1_0         1.1 MB  conda-forge\n",
      "    r-ade4-1.7_18              |    r41he454529_0         5.0 MB  conda-forge\n",
      "    r-ape-5.5                  |    r41h306847c_0         3.1 MB  conda-forge\n",
      "    r-assertthat-0.2.1         |    r41hc72bb7e_2          70 KB  conda-forge\n",
      "    r-backports-1.4.0          |    r41hcfec24a_0         110 KB  conda-forge\n",
      "    r-base-4.1.1               |       hb67fd72_0        24.8 MB  conda-forge\n",
      "    r-bitops-1.0_7             |    r41hcfec24a_0          44 KB  conda-forge\n",
      "    r-brio-1.1.3               |    r41hcfec24a_0          40 KB  conda-forge\n",
      "    r-callr-3.7.0              |    r41hc72bb7e_0         439 KB  conda-forge\n",
      "    r-cli-3.1.0                |    r41h03ef668_0         1.1 MB  conda-forge\n",
      "    r-cluster-2.1.2            |    r41h859d828_0         570 KB  conda-forge\n",
      "    r-codetools-0.2_18         |    r41hc72bb7e_0         106 KB  conda-forge\n",
      "    r-colorspace-2.0_2         |    r41hcfec24a_0         2.6 MB  conda-forge\n",
      "    r-crayon-1.4.2             |    r41hc72bb7e_0         161 KB  conda-forge\n",
      "    r-data.table-1.14.2        |    r41hcfec24a_0         2.0 MB  conda-forge\n",
      "    r-desc-1.4.0               |    r41hc72bb7e_0         557 KB  conda-forge\n",
      "    r-diffobj-0.3.5            |    r41hcfec24a_0         1.0 MB  conda-forge\n",
      "    r-digest-0.6.29            |    r41h03ef668_0         208 KB  conda-forge\n",
      "    r-ellipsis-0.3.2           |    r41hcfec24a_0          42 KB  conda-forge\n",
      "    r-evaluate-0.14            |    r41hc72bb7e_2          81 KB  conda-forge\n",
      "    r-fansi-0.4.2              |    r41hcfec24a_0         198 KB  conda-forge\n",
      "    r-farver-2.1.0             |    r41h03ef668_0         1.5 MB  conda-forge\n",
      "    r-foreach-1.5.1            |    r41hc72bb7e_0         132 KB  conda-forge\n",
      "    r-ggplot2-3.3.5            |    r41hc72bb7e_0         4.0 MB  conda-forge\n",
      "    r-glue-1.5.1               |    r41hcfec24a_0         152 KB  conda-forge\n",
      "    r-gtable-0.3.0             |    r41hc72bb7e_3         423 KB  conda-forge\n",
      "    r-hms-1.1.1                |    r41hc72bb7e_0         106 KB  conda-forge\n",
      "    r-igraph-1.2.9             |    r41he0372cf_0         3.7 MB  conda-forge\n",
      "    r-isoband-0.2.5            |    r41h03ef668_0         1.9 MB  conda-forge\n",
      "    r-iterators-1.0.13         |    r41hc72bb7e_0         339 KB  conda-forge\n",
      "    r-jsonlite-1.7.2           |    r41hcfec24a_0         462 KB  conda-forge\n",
      "    r-labeling-0.4.2           |    r41hc72bb7e_0          67 KB  conda-forge\n",
      "    r-lattice-0.20_45          |    r41hcfec24a_0         1.1 MB  conda-forge\n",
      "    r-lifecycle-1.0.1          |    r41hc72bb7e_0         109 KB  conda-forge\n",
      "    r-magrittr-2.0.1           |    r41hcfec24a_1         211 KB  conda-forge\n",
      "    r-mass-7.3_54              |    r41hcfec24a_0         1.1 MB  conda-forge\n",
      "    r-matrix-1.3_4             |    r41he454529_0         3.9 MB  conda-forge\n",
      "    r-mgcv-1.8_38              |    r41he454529_0         3.0 MB  conda-forge\n",
      "    r-munsell-0.5.0            | r41hc72bb7e_1003         247 KB  conda-forge\n",
      "    r-nlme-3.1_153             |    r41h859d828_0         2.3 MB  conda-forge\n",
      "    r-permute-0.9_5            |    r41h785f33e_3         507 KB  conda-forge\n",
      "    r-pillar-1.6.4             |    r41hc72bb7e_0         991 KB  conda-forge\n",
      "    r-pixmap-0.4_12            |    r41hc72bb7e_0         229 KB  conda-forge\n",
      "    r-pkgconfig-2.0.3          |    r41hc72bb7e_1          25 KB  conda-forge\n",
      "    r-pkgload-1.2.4            |    r41h03ef668_0         172 KB  conda-forge\n",
      "    r-plyr-1.8.6               |    r41h03ef668_1         831 KB  conda-forge\n",
      "    r-praise-1.0.0             | r41hc72bb7e_1004          23 KB  conda-forge\n",
      "    r-prettyunits-1.1.1        |    r41hc72bb7e_1          41 KB  conda-forge\n",
      "    r-processx-3.5.2           |    r41hcfec24a_0         314 KB  conda-forge\n",
      "    r-progress-1.2.2           |    r41hc72bb7e_2          91 KB  conda-forge\n",
      "    r-ps-1.6.0                 |    r41hcfec24a_0         299 KB  conda-forge\n",
      "    r-r6-2.5.1                 |    r41hc72bb7e_0          89 KB  conda-forge\n",
      "    r-rcolorbrewer-1.1_2       | r41h785f33e_1003          64 KB  conda-forge\n",
      "    r-rcpp-1.0.7               |    r41h03ef668_0         2.0 MB  conda-forge\n",
      "    r-rcurl-1.98_1.5           |    r41hcfec24a_0         989 KB  conda-forge\n",
      "    r-rematch2-2.1.2           |    r41hc72bb7e_1          52 KB  conda-forge\n",
      "    r-reshape2-1.4.4           |    r41h03ef668_1         128 KB  conda-forge\n",
      "    r-rlang-0.4.12             |    r41hcfec24a_0         1.1 MB  conda-forge\n",
      "    r-rprojroot-2.0.2          |    r41hc72bb7e_0         110 KB  conda-forge\n",
      "    r-rstudioapi-0.13          |    r41hc72bb7e_0         281 KB  conda-forge\n",
      "    r-scales-1.1.1             |    r41hc72bb7e_0         557 KB  conda-forge\n",
      "    r-sp-1.4_6                 |    r41hcfec24a_0         2.2 MB  conda-forge\n",
      "    r-stringi-1.7.6            |    r41hcabe038_0         886 KB  conda-forge\n",
      "    r-stringr-1.4.0            |    r41hc72bb7e_2         210 KB  conda-forge\n",
      "    r-survival-3.2_13          |    r41hcfec24a_0         6.5 MB  conda-forge\n",
      "    r-testthat-3.1.1           |    r41h03ef668_0         1.6 MB  conda-forge\n",
      "    r-tibble-3.1.6             |    r41hcfec24a_0         774 KB  conda-forge\n",
      "    r-utf8-1.2.2               |    r41hcfec24a_0         162 KB  conda-forge\n",
      "    r-vctrs-0.3.8              |    r41hcfec24a_1         1.1 MB  conda-forge\n",
      "    r-vegan-2.5_7              |    r41h52d45c5_0         3.4 MB  conda-forge\n",
      "    r-viridislite-0.4.0        |    r41hc72bb7e_0         1.3 MB  conda-forge\n",
      "    r-waldo-0.3.1              |    r41hc72bb7e_0         103 KB  conda-forge\n",
      "    r-withr-2.4.3              |    r41hc72bb7e_0         225 KB  conda-forge\n",
      "    sed-4.8                    |       he412f7d_0         264 KB  conda-forge\n",
      "    sysroot_linux-64-2.12      |      he073ed8_15        31.4 MB  conda-forge\n",
      "    tktable-2.10               |       hb7b940f_3          89 KB  conda-forge\n",
      "    xorg-libxt-1.2.1           |       h7f98852_2         375 KB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:       271.2 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  _r-mutex           conda-forge/noarch::_r-mutex-1.0.1-anacondar_1\n",
      "  binutils_impl_lin~ conda-forge/linux-64::binutils_impl_linux-64-2.36.1-h193b22a_2\n",
      "  binutils_linux-64  conda-forge/linux-64::binutils_linux-64-2.36-hf3e587d_2\n",
      "  bioconductor-biob~ bioconda/linux-64::bioconductor-biobase-2.54.0-r41hd029910_0\n",
      "  bioconductor-bioc~ bioconda/noarch::bioconductor-biocgenerics-0.40.0-r41hdfd78af_0\n",
      "  bioconductor-biom~ bioconda/noarch::bioconductor-biomformat-1.22.0-r41hdfd78af_0\n",
      "  bioconductor-bios~ bioconda/linux-64::bioconductor-biostrings-2.62.0-r41hd029910_0\n",
      "  bioconductor-geno~ bioconda/noarch::bioconductor-genomeinfodb-1.30.0-r41hdfd78af_0\n",
      "  bioconductor-geno~ bioconda/noarch::bioconductor-genomeinfodbdata-1.2.7-r41hdfd78af_1\n",
      "  bioconductor-iran~ bioconda/linux-64::bioconductor-iranges-2.28.0-r41hd029910_0\n",
      "  bioconductor-mult~ bioconda/linux-64::bioconductor-multtest-2.50.0-r41hd029910_0\n",
      "  bioconductor-phyl~ bioconda/noarch::bioconductor-phyloseq-1.38.0-r41hdfd78af_0\n",
      "  bioconductor-rhdf5 bioconda/linux-64::bioconductor-rhdf5-2.38.0-r41hfe70e90_1\n",
      "  bioconductor-rhdf~ bioconda/linux-64::bioconductor-rhdf5filters-1.6.0-r41h399db7b_0\n",
      "  bioconductor-rhdf~ bioconda/linux-64::bioconductor-rhdf5lib-1.16.0-r41hd029910_0\n",
      "  bioconductor-s4ve~ bioconda/linux-64::bioconductor-s4vectors-0.32.0-r41hd029910_0\n",
      "  bioconductor-xvec~ bioconda/linux-64::bioconductor-xvector-0.34.0-r41hd029910_0\n",
      "  bioconductor-zlib~ bioconda/linux-64::bioconductor-zlibbioc-1.40.0-r41hd029910_0\n",
      "  bwidget            conda-forge/linux-64::bwidget-1.9.14-ha770c72_1\n",
      "  gcc_impl_linux-64  conda-forge/linux-64::gcc_impl_linux-64-9.4.0-h03d3576_11\n",
      "  gcc_linux-64       conda-forge/linux-64::gcc_linux-64-9.4.0-h391b98a_2\n",
      "  gfortran_impl_lin~ conda-forge/linux-64::gfortran_impl_linux-64-9.4.0-h0003116_11\n",
      "  gfortran_linux-64  conda-forge/linux-64::gfortran_linux-64-9.4.0-hf0ab688_2\n",
      "  gxx_impl_linux-64  conda-forge/linux-64::gxx_impl_linux-64-9.4.0-h03d3576_11\n",
      "  gxx_linux-64       conda-forge/linux-64::gxx_linux-64-9.4.0-h0316aca_2\n",
      "  kernel-headers_li~ conda-forge/noarch::kernel-headers_linux-64-2.6.32-he073ed8_15\n",
      "  libgcc-devel_linu~ conda-forge/linux-64::libgcc-devel_linux-64-9.4.0-hd854feb_11\n",
      "  libsanitizer       conda-forge/linux-64::libsanitizer-9.4.0-h79bfe98_11\n",
      "  libstdcxx-devel_l~ conda-forge/linux-64::libstdcxx-devel_linux-64-9.4.0-hd854feb_11\n",
      "  make               conda-forge/linux-64::make-4.3-hd18ef5c_1\n",
      "  pcre2              conda-forge/linux-64::pcre2-10.37-h032f7d1_0\n",
      "  r-ade4             conda-forge/linux-64::r-ade4-1.7_18-r41he454529_0\n",
      "  r-ape              conda-forge/linux-64::r-ape-5.5-r41h306847c_0\n",
      "  r-assertthat       conda-forge/noarch::r-assertthat-0.2.1-r41hc72bb7e_2\n",
      "  r-backports        conda-forge/linux-64::r-backports-1.4.0-r41hcfec24a_0\n",
      "  r-base             conda-forge/linux-64::r-base-4.1.1-hb67fd72_0\n",
      "  r-bitops           conda-forge/linux-64::r-bitops-1.0_7-r41hcfec24a_0\n",
      "  r-brio             conda-forge/linux-64::r-brio-1.1.3-r41hcfec24a_0\n",
      "  r-callr            conda-forge/noarch::r-callr-3.7.0-r41hc72bb7e_0\n",
      "  r-cli              conda-forge/linux-64::r-cli-3.1.0-r41h03ef668_0\n",
      "  r-cluster          conda-forge/linux-64::r-cluster-2.1.2-r41h859d828_0\n",
      "  r-codetools        conda-forge/noarch::r-codetools-0.2_18-r41hc72bb7e_0\n",
      "  r-colorspace       conda-forge/linux-64::r-colorspace-2.0_2-r41hcfec24a_0\n",
      "  r-crayon           conda-forge/noarch::r-crayon-1.4.2-r41hc72bb7e_0\n",
      "  r-data.table       conda-forge/linux-64::r-data.table-1.14.2-r41hcfec24a_0\n",
      "  r-desc             conda-forge/noarch::r-desc-1.4.0-r41hc72bb7e_0\n",
      "  r-diffobj          conda-forge/linux-64::r-diffobj-0.3.5-r41hcfec24a_0\n",
      "  r-digest           conda-forge/linux-64::r-digest-0.6.29-r41h03ef668_0\n",
      "  r-ellipsis         conda-forge/linux-64::r-ellipsis-0.3.2-r41hcfec24a_0\n",
      "  r-evaluate         conda-forge/noarch::r-evaluate-0.14-r41hc72bb7e_2\n",
      "  r-fansi            conda-forge/linux-64::r-fansi-0.4.2-r41hcfec24a_0\n",
      "  r-farver           conda-forge/linux-64::r-farver-2.1.0-r41h03ef668_0\n",
      "  r-foreach          conda-forge/noarch::r-foreach-1.5.1-r41hc72bb7e_0\n",
      "  r-ggplot2          conda-forge/noarch::r-ggplot2-3.3.5-r41hc72bb7e_0\n",
      "  r-glue             conda-forge/linux-64::r-glue-1.5.1-r41hcfec24a_0\n",
      "  r-gtable           conda-forge/noarch::r-gtable-0.3.0-r41hc72bb7e_3\n",
      "  r-hms              conda-forge/noarch::r-hms-1.1.1-r41hc72bb7e_0\n",
      "  r-igraph           conda-forge/linux-64::r-igraph-1.2.9-r41he0372cf_0\n",
      "  r-isoband          conda-forge/linux-64::r-isoband-0.2.5-r41h03ef668_0\n",
      "  r-iterators        conda-forge/noarch::r-iterators-1.0.13-r41hc72bb7e_0\n",
      "  r-jsonlite         conda-forge/linux-64::r-jsonlite-1.7.2-r41hcfec24a_0\n",
      "  r-labeling         conda-forge/noarch::r-labeling-0.4.2-r41hc72bb7e_0\n",
      "  r-lattice          conda-forge/linux-64::r-lattice-0.20_45-r41hcfec24a_0\n",
      "  r-lifecycle        conda-forge/noarch::r-lifecycle-1.0.1-r41hc72bb7e_0\n",
      "  r-magrittr         conda-forge/linux-64::r-magrittr-2.0.1-r41hcfec24a_1\n",
      "  r-mass             conda-forge/linux-64::r-mass-7.3_54-r41hcfec24a_0\n",
      "  r-matrix           conda-forge/linux-64::r-matrix-1.3_4-r41he454529_0\n",
      "  r-mgcv             conda-forge/linux-64::r-mgcv-1.8_38-r41he454529_0\n",
      "  r-munsell          conda-forge/noarch::r-munsell-0.5.0-r41hc72bb7e_1003\n",
      "  r-nlme             conda-forge/linux-64::r-nlme-3.1_153-r41h859d828_0\n",
      "  r-permute          conda-forge/noarch::r-permute-0.9_5-r41h785f33e_3\n",
      "  r-pillar           conda-forge/noarch::r-pillar-1.6.4-r41hc72bb7e_0\n",
      "  r-pixmap           conda-forge/noarch::r-pixmap-0.4_12-r41hc72bb7e_0\n",
      "  r-pkgconfig        conda-forge/noarch::r-pkgconfig-2.0.3-r41hc72bb7e_1\n",
      "  r-pkgload          conda-forge/linux-64::r-pkgload-1.2.4-r41h03ef668_0\n",
      "  r-plyr             conda-forge/linux-64::r-plyr-1.8.6-r41h03ef668_1\n",
      "  r-praise           conda-forge/noarch::r-praise-1.0.0-r41hc72bb7e_1004\n",
      "  r-prettyunits      conda-forge/noarch::r-prettyunits-1.1.1-r41hc72bb7e_1\n",
      "  r-processx         conda-forge/linux-64::r-processx-3.5.2-r41hcfec24a_0\n",
      "  r-progress         conda-forge/noarch::r-progress-1.2.2-r41hc72bb7e_2\n",
      "  r-ps               conda-forge/linux-64::r-ps-1.6.0-r41hcfec24a_0\n",
      "  r-r6               conda-forge/noarch::r-r6-2.5.1-r41hc72bb7e_0\n",
      "  r-rcolorbrewer     conda-forge/noarch::r-rcolorbrewer-1.1_2-r41h785f33e_1003\n",
      "  r-rcpp             conda-forge/linux-64::r-rcpp-1.0.7-r41h03ef668_0\n",
      "  r-rcurl            conda-forge/linux-64::r-rcurl-1.98_1.5-r41hcfec24a_0\n",
      "  r-rematch2         conda-forge/noarch::r-rematch2-2.1.2-r41hc72bb7e_1\n",
      "  r-reshape2         conda-forge/linux-64::r-reshape2-1.4.4-r41h03ef668_1\n",
      "  r-rlang            conda-forge/linux-64::r-rlang-0.4.12-r41hcfec24a_0\n",
      "  r-rprojroot        conda-forge/noarch::r-rprojroot-2.0.2-r41hc72bb7e_0\n",
      "  r-rstudioapi       conda-forge/noarch::r-rstudioapi-0.13-r41hc72bb7e_0\n",
      "  r-scales           conda-forge/noarch::r-scales-1.1.1-r41hc72bb7e_0\n",
      "  r-sp               conda-forge/linux-64::r-sp-1.4_6-r41hcfec24a_0\n",
      "  r-stringi          conda-forge/linux-64::r-stringi-1.7.6-r41hcabe038_0\n",
      "  r-stringr          conda-forge/noarch::r-stringr-1.4.0-r41hc72bb7e_2\n",
      "  r-survival         conda-forge/linux-64::r-survival-3.2_13-r41hcfec24a_0\n",
      "  r-testthat         conda-forge/linux-64::r-testthat-3.1.1-r41h03ef668_0\n",
      "  r-tibble           conda-forge/linux-64::r-tibble-3.1.6-r41hcfec24a_0\n",
      "  r-utf8             conda-forge/linux-64::r-utf8-1.2.2-r41hcfec24a_0\n",
      "  r-vctrs            conda-forge/linux-64::r-vctrs-0.3.8-r41hcfec24a_1\n",
      "  r-vegan            conda-forge/linux-64::r-vegan-2.5_7-r41h52d45c5_0\n",
      "  r-viridislite      conda-forge/noarch::r-viridislite-0.4.0-r41hc72bb7e_0\n",
      "  r-waldo            conda-forge/noarch::r-waldo-0.3.1-r41hc72bb7e_0\n",
      "  r-withr            conda-forge/noarch::r-withr-2.4.3-r41hc72bb7e_0\n",
      "  sed                conda-forge/linux-64::sed-4.8-he412f7d_0\n",
      "  sysroot_linux-64   conda-forge/noarch::sysroot_linux-64-2.12-he073ed8_15\n",
      "  tktable            conda-forge/linux-64::tktable-2.10-hb7b940f_3\n",
      "  xorg-libxt         conda-forge/linux-64::xorg-libxt-1.2.1-h7f98852_2\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "r-crayon-1.4.2       | 161 KB    | ##################################### | 100% \n",
      "pcre2-10.37          | 1.1 MB    | ##################################### | 100% \n",
      "bioconductor-genomei | 4.1 MB    | ##################################### | 100% \n",
      "r-digest-0.6.29      | 208 KB    | ##################################### | 100% \n",
      "r-fansi-0.4.2        | 198 KB    | ##################################### | 100% \n",
      "r-base-4.1.1         | 24.8 MB   | ##################################### | 100% \n",
      "xorg-libxt-1.2.1     | 375 KB    | ##################################### | 100% \n",
      "bioconductor-biocgen | 631 KB    | ##################################### | 100% \n",
      "r-mass-7.3_54        | 1.1 MB    | ##################################### | 100% \n",
      "r-survival-3.2_13    | 6.5 MB    | ##################################### | 100% \n",
      "r-foreach-1.5.1      | 132 KB    | ##################################### | 100% \n",
      "r-waldo-0.3.1        | 103 KB    | ##################################### | 100% \n",
      "make-4.3             | 507 KB    | ##################################### | 100% \n",
      "gxx_impl_linux-64-9. | 10.6 MB   | ##################################### | 100% \n",
      "r-utf8-1.2.2         | 162 KB    | ##################################### | 100% \n",
      "r-rstudioapi-0.13    | 281 KB    | ##################################### | 100% \n",
      "r-ggplot2-3.3.5      | 4.0 MB    | ##################################### | 100% \n",
      "r-hms-1.1.1          | 106 KB    | ##################################### | 100% \n",
      "bioconductor-rhdf5-2 | 2.8 MB    | ##################################### | 100% \n",
      "gfortran_impl_linux- | 14.6 MB   | ##################################### | 100% \n",
      "r-pixmap-0.4_12      | 229 KB    | ##################################### | 100% \n",
      "r-pkgload-1.2.4      | 172 KB    | ##################################### | 100% \n",
      "r-gtable-0.3.0       | 423 KB    | ##################################### | 100% \n",
      "r-rlang-0.4.12       | 1.1 MB    | ##################################### | 100% \n",
      "r-rematch2-2.1.2     | 52 KB     | ##################################### | 100% \n",
      "bioconductor-biostri | 13.9 MB   | ##################################### | 100% \n",
      "r-colorspace-2.0_2   | 2.6 MB    | ##################################### | 100% \n",
      "bioconductor-iranges | 2.4 MB    | ##################################### | 100% \n",
      "bwidget-1.9.14       | 120 KB    | ##################################### | 100% \n",
      "r-diffobj-0.3.5      | 1.0 MB    | ##################################### | 100% \n",
      "r-codetools-0.2_18   | 106 KB    | ##################################### | 100% \n",
      "libstdcxx-devel_linu | 9.9 MB    | ##################################### | 100% \n",
      "r-processx-3.5.2     | 314 KB    | ##################################### | 100% \n",
      "r-ellipsis-0.3.2     | 42 KB     | ##################################### | 100% \n",
      "bioconductor-biobase | 2.3 MB    | ##################################### | 100% \n",
      "r-scales-1.1.1       | 557 KB    | ##################################### | 100% \n",
      "r-data.table-1.14.2  | 2.0 MB    | ##################################### | 100% \n",
      "r-bitops-1.0_7       | 44 KB     | ##################################### | 100% \n",
      "r-assertthat-0.2.1   | 70 KB     | ##################################### | 100% \n",
      "binutils_linux-64-2. | 23 KB     | ##################################### | 100% \n",
      "tktable-2.10         | 89 KB     | ##################################### | 100% \n",
      "bioconductor-rhdf5li | 3.9 MB    | ##################################### | 100% \n",
      "_r-mutex-1.0.1       | 3 KB      | ##################################### | 100% \n",
      "r-prettyunits-1.1.1  | 41 KB     | ##################################### | 100% \n",
      "r-rcolorbrewer-1.1_2 | 64 KB     | ##################################### | 100% \n",
      "sed-4.8              | 264 KB    | ##################################### | 100% \n",
      "r-mgcv-1.8_38        | 3.0 MB    | ##################################### | 100% \n",
      "r-ps-1.6.0           | 299 KB    | ##################################### | 100% \n",
      "r-jsonlite-1.7.2     | 462 KB    | ##################################### | 100% \n",
      "gfortran_linux-64-9. | 24 KB     | ##################################### | 100% \n",
      "r-stringr-1.4.0      | 210 KB    | ##################################### | 100% \n",
      "r-permute-0.9_5      | 507 KB    | ##################################### | 100% \n",
      "r-backports-1.4.0    | 110 KB    | ##################################### | 100% \n",
      "bioconductor-multtes | 889 KB    | ##################################### | 100% \n",
      "r-callr-3.7.0        | 439 KB    | ##################################### | 100% \n",
      "r-vegan-2.5_7        | 3.4 MB    | ##################################### | 100% \n",
      "r-brio-1.1.3         | 40 KB     | ##################################### | 100% \n",
      "kernel-headers_linux | 707 KB    | ##################################### | 100% \n",
      "r-viridislite-0.4.0  | 1.3 MB    | ##################################### | 100% \n",
      "r-lifecycle-1.0.1    | 109 KB    | ##################################### | 100% \n",
      "r-isoband-0.2.5      | 1.9 MB    | ##################################### | 100% \n",
      "r-nlme-3.1_153       | 2.3 MB    | ##################################### | 100% \n",
      "r-ade4-1.7_18        | 5.0 MB    | ##################################### | 100% \n",
      "gcc_linux-64-9.4.0   | 24 KB     | ##################################### | 100% \n",
      "sysroot_linux-64-2.1 | 31.4 MB   | ##################################### | 100% \n",
      "r-r6-2.5.1           | 89 KB     | ##################################### | 100% \n",
      "binutils_impl_linux- | 10.4 MB   | ##################################### | 100% \n",
      "gcc_impl_linux-64-9. | 45.6 MB   | ##################################### | 100% \n",
      "r-glue-1.5.1         | 152 KB    | ##################################### | 100% \n",
      "r-pillar-1.6.4       | 991 KB    | ##################################### | 100% \n",
      "bioconductor-genomei | 7 KB      | ##################################### | 100% \n",
      "bioconductor-zlibbio | 114 KB    | ##################################### | 100% \n",
      "r-praise-1.0.0       | 23 KB     | ##################################### | 100% \n",
      "bioconductor-s4vecto | 2.2 MB    | ##################################### | 100% \n",
      "r-withr-2.4.3        | 225 KB    | ##################################### | 100% \n",
      "r-rprojroot-2.0.2    | 110 KB    | ##################################### | 100% \n",
      "r-labeling-0.4.2     | 67 KB     | ##################################### | 100% \n",
      "r-iterators-1.0.13   | 339 KB    | ##################################### | 100% \n",
      "r-cli-3.1.0          | 1.1 MB    | ##################################### | 100% \n",
      "libgcc-devel_linux-6 | 4.0 MB    | ##################################### | 100% \n",
      "libsanitizer-9.4.0   | 6.9 MB    | ##################################### | 100% \n",
      "r-magrittr-2.0.1     | 211 KB    | ##################################### | 100% \n",
      "r-farver-2.1.0       | 1.5 MB    | ##################################### | 100% \n",
      "r-progress-1.2.2     | 91 KB     | ##################################### | 100% \n",
      "bioconductor-rhdf5fi | 664 KB    | ##################################### | 100% \n",
      "r-testthat-3.1.1     | 1.6 MB    | ##################################### | 100% \n",
      "r-rcpp-1.0.7         | 2.0 MB    | ##################################### | 100% \n",
      "r-cluster-2.1.2      | 570 KB    | ##################################### | 100% \n",
      "r-plyr-1.8.6         | 831 KB    | ##################################### | 100% \n",
      "gxx_linux-64-9.4.0   | 24 KB     | ##################################### | 100% \n",
      "r-ape-5.5            | 3.1 MB    | ##################################### | 100% \n",
      "r-rcurl-1.98_1.5     | 989 KB    | ##################################### | 100% \n",
      "r-stringi-1.7.6      | 886 KB    | ##################################### | 100% \n",
      "bioconductor-biomfor | 542 KB    | ##################################### | 100% \n",
      "r-munsell-0.5.0      | 247 KB    | ##################################### | 100% \n",
      "bioconductor-xvector | 726 KB    | ##################################### | 100% \n",
      "r-desc-1.4.0         | 557 KB    | ##################################### | 100% \n",
      "r-sp-1.4_6           | 2.2 MB    | ##################################### | 100% \n",
      "r-vctrs-0.3.8        | 1.1 MB    | ##################################### | 100% \n",
      "r-igraph-1.2.9       | 3.7 MB    | ##################################### | 100% \n",
      "bioconductor-phylose | 5.7 MB    | ##################################### | 100% \n",
      "r-tibble-3.1.6       | 774 KB    | ##################################### | 100% \n",
      "r-pkgconfig-2.0.3    | 25 KB     | ##################################### | 100% \n",
      "r-lattice-0.20_45    | 1.1 MB    | ##################################### | 100% \n",
      "r-evaluate-0.14      | 81 KB     | ##################################### | 100% \n",
      "r-matrix-1.3_4       | 3.9 MB    | ##################################### | 100% \n",
      "r-reshape2-1.4.4     | 128 KB    | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c bioconda bioconductor-phyloseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "27f6a64d-581f-4854-b55c-a55c94919baa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c conda-forge r-ggplot2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "74eb8974-1862-488b-8973-5bab9e162962",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c conda-forge r-reshape2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "f1d13f3b-fddc-448a-bcb0-6ddbaa7b83a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c conda-forge r-vegan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "81fa36e9-b528-44bf-bf5b-24e45c05e4d7",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - r-dplyr\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    r-dplyr-1.0.7              |    r41h03ef668_0         1.1 MB  conda-forge\n",
      "    r-generics-0.1.1           |    r41hc72bb7e_0          89 KB  conda-forge\n",
      "    r-purrr-0.3.4              |    r41hcfec24a_1         410 KB  conda-forge\n",
      "    r-tidyselect-1.1.1         |    r41hc72bb7e_0         195 KB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:         1.8 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  r-dplyr            conda-forge/linux-64::r-dplyr-1.0.7-r41h03ef668_0\n",
      "  r-generics         conda-forge/noarch::r-generics-0.1.1-r41hc72bb7e_0\n",
      "  r-purrr            conda-forge/linux-64::r-purrr-0.3.4-r41hcfec24a_1\n",
      "  r-tidyselect       conda-forge/linux-64::r-tidyselect-1.1.1-r41hc72bb7e_0\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "r-dplyr-1.0.7        | 1.1 MB    | ##################################### | 100% \n",
      "r-purrr-0.3.4        | 410 KB    | ##################################### | 100% \n",
      "r-tidyselect-1.1.1   | 195 KB    | ##################################### | 100% \n",
      "r-generics-0.1.1     | 89 KB     | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c conda-forge r-dplyr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "0ea07855-2c6f-4191-b302-985fff765b25",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "# All requested packages already installed.\n",
      "\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c conda-forge r-rcolorbrewer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "01656ce1-35d3-4c4a-a9e9-8c9c2e26f57d",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting package metadata (current_repodata.json): done\n",
      "Solving environment: done\n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /opt/conda\n",
      "\n",
      "  added / updated specs:\n",
      "    - r-png\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    r-png-0.1_7                | r41hcfec24a_1004          58 KB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:          58 KB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  r-png              conda-forge/linux-64::r-png-0.1_7-r41hcfec24a_1004\n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages\n",
      "r-png-0.1_7          | 58 KB     | ##################################### | 100% \n",
      "Preparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda install -c conda-forge r-png"
   ]
  }
 ],
 "metadata": {
  "environment": {
   "kernel": "conda-root-py",
   "name": "common-cpu.m87",
   "type": "gcloud",
   "uri": "gcr.io/deeplearning-platform-release/base-cpu:m87"
  },
  "kernelspec": {
   "display_name": "Python [conda env:root] *",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
